<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.0.36">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Myelin insulation as a risk factor for axonal degeneration in autoimmune demyelinating disease. Analysis. Profiling of CD45+ immune cells from wt and hMBP SpC 4 days post-EAE onset by single-cell RNA sequencing</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="EAE_IC_files/libs/clipboard/clipboard.min.js"></script>
<script src="EAE_IC_files/libs/quarto-html/quarto.js"></script>
<script src="EAE_IC_files/libs/quarto-html/popper.min.js"></script>
<script src="EAE_IC_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="EAE_IC_files/libs/quarto-html/anchor.min.js"></script>
<link href="EAE_IC_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="EAE_IC_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="EAE_IC_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="EAE_IC_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="EAE_IC_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#scrnaseq-of-cd45-immune-cells-wt-vs-hmbp-in-peak-eae" id="toc-scrnaseq-of-cd45-immune-cells-wt-vs-hmbp-in-peak-eae" class="nav-link active" data-scroll-target="#scrnaseq-of-cd45-immune-cells-wt-vs-hmbp-in-peak-eae">scRNAseq of CD45+ Immune cells WT vs hMBP in peak EAE</a>
  <ul class="collapse">
  <li><a href="#initialization" id="toc-initialization" class="nav-link" data-scroll-target="#initialization">Initialization</a>
  <ul class="collapse">
  <li><a href="#environment-preparation" id="toc-environment-preparation" class="nav-link" data-scroll-target="#environment-preparation">Environment preparation</a></li>
  <li><a href="#document-setup" id="toc-document-setup" class="nav-link" data-scroll-target="#document-setup">Document setup</a></li>
  <li><a href="#attach-required-packages" id="toc-attach-required-packages" class="nav-link" data-scroll-target="#attach-required-packages">Attach required packages</a></li>
  <li><a href="#define-custom-functions" id="toc-define-custom-functions" class="nav-link" data-scroll-target="#define-custom-functions">Define custom functions</a></li>
  <li><a href="#define-input-and-output-directories" id="toc-define-input-and-output-directories" class="nav-link" data-scroll-target="#define-input-and-output-directories">Define input and output directories</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#qc-of-raw-data" id="toc-qc-of-raw-data" class="nav-link" data-scroll-target="#qc-of-raw-data">QC of raw data</a>
  <ul class="collapse">
  <li><a href="#density-of-counts" id="toc-density-of-counts" class="nav-link" data-scroll-target="#density-of-counts">Density of Counts</a></li>
  <li><a href="#density-of-genes" id="toc-density-of-genes" class="nav-link" data-scroll-target="#density-of-genes">Density of Genes</a></li>
  <li><a href="#summary" id="toc-summary" class="nav-link" data-scroll-target="#summary">Summary</a></li>
  <li><a href="#filtering" id="toc-filtering" class="nav-link" data-scroll-target="#filtering">Filtering</a></li>
  </ul></li>
  <li><a href="#dimensional-reduction" id="toc-dimensional-reduction" class="nav-link" data-scroll-target="#dimensional-reduction">Dimensional reduction</a>
  <ul class="collapse">
  <li><a href="#normalization" id="toc-normalization" class="nav-link" data-scroll-target="#normalization">Normalization</a></li>
  <li><a href="#variable-genes" id="toc-variable-genes" class="nav-link" data-scroll-target="#variable-genes">Variable genes</a></li>
  <li><a href="#pca" id="toc-pca" class="nav-link" data-scroll-target="#pca">PCA</a></li>
  <li><a href="#umap" id="toc-umap" class="nav-link" data-scroll-target="#umap">UMAP</a></li>
  <li><a href="#unsupervised-clustering" id="toc-unsupervised-clustering" class="nav-link" data-scroll-target="#unsupervised-clustering">Unsupervised clustering</a></li>
  </ul></li>
  <li><a href="#known-markers" id="toc-known-markers" class="nav-link" data-scroll-target="#known-markers">Known markers</a>
  <ul class="collapse">
  <li><a href="#cd45-immune-cells" id="toc-cd45-immune-cells" class="nav-link" data-scroll-target="#cd45-immune-cells">CD45+ immune cells</a></li>
  <li><a href="#cd14-monocytesmacrophagesmicroglia" id="toc-cd14-monocytesmacrophagesmicroglia" class="nav-link" data-scroll-target="#cd14-monocytesmacrophagesmicroglia">CD14+ Monocytes/Macrophages/Microglia</a></li>
  <li><a href="#cd3-t-cells" id="toc-cd3-t-cells" class="nav-link" data-scroll-target="#cd3-t-cells">CD3+ T-cells</a></li>
  <li><a href="#cd19-b-cells" id="toc-cd19-b-cells" class="nav-link" data-scroll-target="#cd19-b-cells">CD19+ B-cells</a></li>
  <li><a href="#conventional-dcs" id="toc-conventional-dcs" class="nav-link" data-scroll-target="#conventional-dcs">Conventional DCs</a></li>
  <li><a href="#siglec-h-plasmacytoid-dcs" id="toc-siglec-h-plasmacytoid-dcs" class="nav-link" data-scroll-target="#siglec-h-plasmacytoid-dcs">Siglec-H+ plasmacytoid DCs</a></li>
  <li><a href="#granulocytes" id="toc-granulocytes" class="nav-link" data-scroll-target="#granulocytes">Granulocytes</a></li>
  <li><a href="#identifying-remaining-clusters" id="toc-identifying-remaining-clusters" class="nav-link" data-scroll-target="#identifying-remaining-clusters">Identifying remaining clusters</a></li>
  <li><a href="#assigning-preliminary-cell-type-labels." id="toc-assigning-preliminary-cell-type-labels." class="nav-link" data-scroll-target="#assigning-preliminary-cell-type-labels.">Assigning preliminary cell type labels.</a></li>
  <li><a href="#cell-cycle" id="toc-cell-cycle" class="nav-link" data-scroll-target="#cell-cycle">Cell cycle</a></li>
  </ul></li>
  <li><a href="#subclustering-analysis" id="toc-subclustering-analysis" class="nav-link" data-scroll-target="#subclustering-analysis">Subclustering analysis</a>
  <ul class="collapse">
  <li><a href="#macrophages" id="toc-macrophages" class="nav-link" data-scroll-target="#macrophages">Macrophages</a>
  <ul class="collapse">
  <li><a href="#integration-reanalysis-of-macrophages" id="toc-integration-reanalysis-of-macrophages" class="nav-link" data-scroll-target="#integration-reanalysis-of-macrophages">Integration &amp; reanalysis of Macrophages</a></li>
  <li><a href="#identify-contaminating-cell-types" id="toc-identify-contaminating-cell-types" class="nav-link" data-scroll-target="#identify-contaminating-cell-types">Identify contaminating cell types</a></li>
  <li><a href="#identification-of-specific-macrophage-cell-states" id="toc-identification-of-specific-macrophage-cell-states" class="nav-link" data-scroll-target="#identification-of-specific-macrophage-cell-states">Identification of specific macrophage cell states</a></li>
  <li><a href="#rename-subtypes" id="toc-rename-subtypes" class="nav-link" data-scroll-target="#rename-subtypes">rename subtypes</a></li>
  </ul></li>
  <li><a href="#microglia" id="toc-microglia" class="nav-link" data-scroll-target="#microglia">Microglia</a>
  <ul class="collapse">
  <li><a href="#integration-of-microglia" id="toc-integration-of-microglia" class="nav-link" data-scroll-target="#integration-of-microglia">Integration of Microglia</a></li>
  <li><a href="#check-for-contamination-in-microglia-sample" id="toc-check-for-contamination-in-microglia-sample" class="nav-link" data-scroll-target="#check-for-contamination-in-microglia-sample">check for contamination in Microglia sample</a></li>
  <li><a href="#clean-contamiting-cells" id="toc-clean-contamiting-cells" class="nav-link" data-scroll-target="#clean-contamiting-cells">clean contamiting cells</a></li>
  <li><a href="#identification-of-specific-microglia-cell-states" id="toc-identification-of-specific-microglia-cell-states" class="nav-link" data-scroll-target="#identification-of-specific-microglia-cell-states">Identification of specific Microglia cell states</a></li>
  <li><a href="#assign-mg-type-names" id="toc-assign-mg-type-names" class="nav-link" data-scroll-target="#assign-mg-type-names">assign mg type names</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#figures" id="toc-figures" class="nav-link" data-scroll-target="#figures">Figures</a>
  <ul class="collapse">
  <li><a href="#load-datasets" id="toc-load-datasets" class="nav-link" data-scroll-target="#load-datasets">Load datasets</a></li>
  <li><a href="#initialize-figures" id="toc-initialize-figures" class="nav-link" data-scroll-target="#initialize-figures">Initialize figures</a></li>
  <li><a href="#overview" id="toc-overview" class="nav-link" data-scroll-target="#overview">Overview</a></li>
  <li><a href="#gene-expression-plots" id="toc-gene-expression-plots" class="nav-link" data-scroll-target="#gene-expression-plots">Gene expression plots</a>
  <ul class="collapse">
  <li><a href="#microglia-1" id="toc-microglia-1" class="nav-link" data-scroll-target="#microglia-1">Microglia</a></li>
  <li><a href="#t-cells" id="toc-t-cells" class="nav-link" data-scroll-target="#t-cells">T-cells</a></li>
  <li><a href="#cytokines" id="toc-cytokines" class="nav-link" data-scroll-target="#cytokines">Cytokines</a></li>
  <li><a href="#macrophages-1" id="toc-macrophages-1" class="nav-link" data-scroll-target="#macrophages-1">Macrophages</a></li>
  <li><a href="#output-fig.-x" id="toc-output-fig.-x" class="nav-link" data-scroll-target="#output-fig.-x">Output Fig. X</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#extended-figures" id="toc-extended-figures" class="nav-link" data-scroll-target="#extended-figures">Extended Figures</a>
  <ul class="collapse">
  <li><a href="#qc" id="toc-qc" class="nav-link" data-scroll-target="#qc">QC</a></li>
  <li><a href="#pca-1" id="toc-pca-1" class="nav-link" data-scroll-target="#pca-1">PCA</a></li>
  <li><a href="#celltypemarkers" id="toc-celltypemarkers" class="nav-link" data-scroll-target="#celltypemarkers">CellTypeMarkers</a></li>
  <li><a href="#subcluster-analysis" id="toc-subcluster-analysis" class="nav-link" data-scroll-target="#subcluster-analysis">Subcluster analysis</a>
  <ul class="collapse">
  <li><a href="#macrophages-fig-x." id="toc-macrophages-fig-x." class="nav-link" data-scroll-target="#macrophages-fig-x.">Macrophages Fig X.</a></li>
  <li><a href="#microglia-fig-x." id="toc-microglia-fig-x." class="nav-link" data-scroll-target="#microglia-fig-x.">Microglia Fig X.</a></li>
  </ul></li>
  </ul></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Myelin insulation as a risk factor for axonal degeneration in autoimmune demyelinating disease. Analysis. Profiling of CD45+ immune cells from wt and hMBP SpC 4 days post-EAE onset by single-cell RNA sequencing</h1>
</div>



<div class="quarto-title-meta">

    
    
  </div>
  

</header>

<section id="scrnaseq-of-cd45-immune-cells-wt-vs-hmbp-in-peak-eae" class="level1">
<h1>scRNAseq of CD45+ Immune cells WT vs hMBP in peak EAE</h1>
<section id="initialization" class="level2">
<h2 class="anchored" data-anchor-id="initialization">Initialization</h2>
<section id="environment-preparation" class="level3">
<h3 class="anchored" data-anchor-id="environment-preparation">Environment preparation</h3>
</section>
<section id="document-setup" class="level3">
<h3 class="anchored" data-anchor-id="document-setup">Document setup</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">message =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">warning =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">error =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="attach-required-packages" class="level3">
<h3 class="anchored" data-anchor-id="attach-required-packages">Attach required packages</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co">#general packages </span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(PCAtools)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="co">#single cell packages </span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SeuratDisk)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scDblFinder)</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="co">#graphics packages</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scCustomize)</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(facefuns)</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(viridis)</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Polychrome)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="define-custom-functions" class="level3">
<h3 class="anchored" data-anchor-id="define-custom-functions">Define custom functions</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co">#function to provide negated set operation</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="st">`</span><span class="at">%!in%</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">Negate</span>(<span class="st">`</span><span class="at">%in%</span><span class="st">`</span>)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="co">#function to calculate relative abundance of individual clusters to a sample </span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>pTable <span class="ot">=</span> <span class="cf">function</span>(seurat_object,cluster_column,split_by_){</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>  x <span class="ot">=</span> <span class="fu">data.frame</span>(</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>        <span class="fu">t</span>(</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>          <span class="fu">t</span>(</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>            <span class="fu">table</span>(</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>              seurat_object<span class="sc">@</span>meta.data[[cluster_column]],</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>              seurat_object<span class="sc">@</span>meta.data[[split_by_]]</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>            )  </span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>          <span class="sc">/</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>            <span class="fu">colSums</span>(</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>              <span class="fu">table</span>(</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>                seurat_object<span class="sc">@</span>meta.data[[cluster_column]],</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>               seurat_object<span class="sc">@</span>meta.data[[split_by_]]</span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>                    )</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>                    )</span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>          <span class="sc">*</span><span class="dv">100</span></span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>          )</span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>                    ) </span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>  <span class="co">#to programmatically rename columns using dplyr evaluate variable content with !! and use special operator :=</span></span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>  x <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(x,</span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>                    <span class="sc">!!</span><span class="at">cluster_column :=</span> Var1,</span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>                    <span class="sc">!!</span><span class="at">split_by_ :=</span> Var2)</span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>  y <span class="ot">=</span> <span class="fu">data.frame</span>(</span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a>    <span class="co">#the levels of the cluster_column e.g. celltypes are the output lavels</span></span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a>        <span class="fu">unique</span>(x[[cluster_column]]), </span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a>    <span class="co">#calculating the log2 of the ratio of frequencies in each of the split_by_ entities is the log2FC</span></span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a>    <span class="co">#with .data[[var_name]] one can programmatically access the columns</span></span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a>        x  <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="fu">log2</span>(</span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a>                            Freq[.data[[split_by_]] <span class="sc">==</span> </span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">levels</span>(</span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">as.factor</span>(</span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a>                                        x[[split_by_]]))[<span class="dv">1</span>]] </span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a>                                                          <span class="sc">/</span> </span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a>                             Freq[.data[[split_by_]] <span class="sc">==</span> </span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">levels</span>(</span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">as.factor</span>(</span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a>                                        x[[split_by_]]))[<span class="dv">2</span>]])))</span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(y) <span class="ot">=</span> <span class="fu">c</span>(cluster_column,<span class="st">'log2FC'</span>)</span>
<span id="cb3-45"><a href="#cb3-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">list</span>(<span class="st">'freq'</span> <span class="ot">=</span> x,</span>
<span id="cb3-46"><a href="#cb3-46" aria-hidden="true" tabindex="-1"></a>              <span class="st">'log2fc'</span> <span class="ot">=</span> y))</span>
<span id="cb3-47"><a href="#cb3-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-48"><a href="#cb3-48" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="define-input-and-output-directories" class="level3">
<h3 class="anchored" data-anchor-id="define-input-and-output-directories">Define input and output directories</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>wd <span class="ot">=</span> <span class="fu">getwd</span>()</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>indir <span class="ot">=</span> <span class="fu">paste0</span>(wd,<span class="st">'/scRNAdata/IC/'</span>)</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(indir,<span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>outdir <span class="ot">=</span> <span class="fu">paste0</span>(wd,<span class="st">'/Output/IC/'</span>)</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(outdir,<span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>figdir <span class="ot">=</span> <span class="fu">paste0</span>(wd,<span class="st">'/figures/IC/'</span>)</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(figdir,<span class="at">recursive =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Reading in raw count matrices output by CellRanger, annotating meta data, merging both hMBP and WT counts into one object. Barcodes with less than 100 reads are discarded as empty droplets.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tag</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>filenameWT <span class="ot">=</span> <span class="fu">paste0</span>(indir,<span class="st">'GSM6914114_MQ_hMBP_EAE_IC_pre_m.h5'</span>)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>filenameHMBP <span class="ot">=</span> <span class="fu">paste0</span>(indir,<span class="st">'GSM6914113_MQ_WT_EAE_IC_pre_m.h5'</span>)</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Reading in Raw 10X count files </span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>rawWT <span class="ot">=</span> <span class="fu">Read10X_h5</span>(filenameWT, <span class="at">use.names =</span> <span class="cn">TRUE</span>, <span class="at">unique.features =</span> <span class="cn">TRUE</span>)</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>rawHMBP <span class="ot">=</span> <span class="fu">Read10X_h5</span>(filenameHMBP, <span class="at">use.names =</span> <span class="cn">TRUE</span>, <span class="at">unique.features =</span> <span class="cn">TRUE</span>)</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="co">#Creating Seurat Objects </span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>sc.taggedWT <span class="ot">=</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span> rawWT, <span class="at">min.cells =</span> <span class="dv">0</span>, <span class="at">min.genes =</span> <span class="dv">200</span>, <span class="at">project =</span> <span class="st">'EAE_IC_WT'</span>)</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>sc.taggedWT<span class="sc">$</span>genotype <span class="ot">=</span> <span class="st">'WT'</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>sc.taggedHMBP <span class="ot">=</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span> rawHMBP, <span class="at">min.cells =</span> <span class="dv">0</span>, <span class="at">min.genes =</span> <span class="dv">200</span>, <span class="at">project =</span> <span class="st">'EAE_IC_HMBP'</span>)</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>sc.taggedHMBP<span class="sc">$</span>genotype <span class="ot">=</span> <span class="st">'hMBP'</span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a><span class="co">#Merging objects</span></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>sc.tagged <span class="ot">=</span> <span class="fu">merge</span>(sc.taggedWT,sc.taggedHMBP)</span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a><span class="co">#filtering empty droplets </span></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>sc.tagged <span class="ot">=</span> sc.tagged <span class="sc">%&gt;%</span> <span class="fu">subset</span>(nCount_RNA <span class="sc">&gt;</span> <span class="dv">100</span>)</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a><span class="co">#factoring sample metadata </span></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>sc.tagged<span class="sc">$</span>sample <span class="ot">=</span> <span class="fu">factor</span>(sc.tagged<span class="sc">$</span>orig.ident,<span class="at">levels =</span> <span class="fu">c</span>(<span class="st">'EAE_IC_WT'</span>,<span class="st">'EAE_IC_HMBP'</span>))</span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>sc.tagged<span class="sc">$</span>genotype <span class="ot">=</span> <span class="fu">factor</span>(sc.tagged<span class="sc">$</span>genotype, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">'WT'</span>,<span class="st">'hMBP'</span>))</span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a><span class="co">#calculating mitochondrial genes</span></span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a>sc.tagged<span class="sc">$</span>mitoRatio <span class="ot">&lt;-</span> <span class="fu">PercentageFeatureSet</span>(<span class="at">object =</span> sc.tagged, <span class="at">features =</span> <span class="fu">rownames</span>(sc.tagged)[<span class="fu">grep</span>(<span class="st">'^mt'</span>,<span class="at">x =</span><span class="fu">rownames</span>(sc.tagged))]</span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a>) <span class="sc">/</span> <span class="dv">100</span></span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a><span class="fu">SaveH5Seurat</span>(sc.tagged,<span class="fu">paste0</span>(outdir,<span class="st">'IC.tagged.h5seurat'</span>),<span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
</section>
<section id="qc-of-raw-data" class="level1">
<h1>QC of raw data</h1>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>cutoff.reads <span class="ot">=</span> <span class="dv">1000</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>cutoff.features <span class="ot">=</span> <span class="dv">500</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>cutoff.mitoRatio <span class="ot">=</span> <span class="fl">0.05</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="density-of-counts" class="level2">
<h2 class="anchored" data-anchor-id="density-of-counts">Density of Counts</h2>
<p>Plotting densities of counts/cell by sample. The line represents the applied cutoff.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tag</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>qc.counts.density <span class="ot">=</span> sc.tagged<span class="sc">@</span>meta.data <span class="sc">%&gt;%</span> </span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">color=</span>sample, <span class="at">x=</span>nCount_RNA, <span class="at">fill=</span> sample)) <span class="sc">+</span> </span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_density</span>(<span class="at">alpha =</span> <span class="fl">0.2</span>) <span class="sc">+</span> </span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_log10</span>() <span class="sc">+</span> </span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">"Cell density"</span>) <span class="sc">+</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> cutoff.reads) <span class="sc">+</span> </span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'none'</span>)</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>qc.counts.violin <span class="ot">=</span> sc.tagged<span class="sc">@</span>meta.data <span class="sc">%&gt;%</span> </span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>sample, <span class="at">y=</span>nCount_RNA, <span class="at">fill=</span> sample, <span class="at">color =</span> sample)) <span class="sc">+</span> </span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_violin</span>(<span class="at">scale =</span> <span class="st">'width'</span>, <span class="at">alpha =</span> <span class="fl">0.2</span>) <span class="sc">+</span> </span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="fl">0.1</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">'black'</span>) <span class="sc">+</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_discrete</span>() <span class="sc">+</span> </span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_log10</span>() <span class="sc">+</span> </span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">width=</span><span class="fl">0.2</span>, <span class="at">outlier.size =</span> <span class="dv">0</span>, <span class="at">fill =</span> <span class="st">'white'</span>, <span class="at">color =</span> <span class="st">'black'</span>, <span class="at">coef =</span> <span class="dv">0</span>, <span class="at">outlier.color =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> cutoff.reads) <span class="sc">+</span> </span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">'Read counts'</span>) <span class="sc">+</span></span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">''</span>) <span class="sc">+</span> </span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'right'</span>)</span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a>qc.counts.density <span class="sc">+</span> qc.counts.violin <span class="sc">+</span> </span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_annotation</span>(</span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">tag_levels =</span> <span class="st">'A'</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/QCdensityCounts-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="density-of-genes" class="level2">
<h2 class="anchored" data-anchor-id="density-of-genes">Density of Genes</h2>
<p>Plotting the density of detected genes/cell by sample. The line represents the applied cutoff.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tag</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>qc.features.density <span class="ot">=</span> sc.tagged<span class="sc">@</span>meta.data <span class="sc">%&gt;%</span> </span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">color=</span>sample, <span class="at">x=</span>nFeature_RNA, <span class="at">fill=</span> sample)) <span class="sc">+</span> </span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_density</span>(<span class="at">alpha =</span> <span class="fl">0.2</span>) <span class="sc">+</span> </span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_log10</span>() <span class="sc">+</span> </span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">"Cell density"</span>) <span class="sc">+</span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> cutoff.features) <span class="sc">+</span> </span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'none'</span>)</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>qc.features.violin <span class="ot">=</span> sc.tagged<span class="sc">@</span>meta.data <span class="sc">%&gt;%</span> </span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>sample, <span class="at">y=</span>nFeature_RNA, <span class="at">fill=</span> sample, <span class="at">color =</span> sample)) <span class="sc">+</span> </span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_violin</span>(<span class="at">scale =</span> <span class="st">'width'</span>, <span class="at">alpha =</span> <span class="fl">0.2</span>) <span class="sc">+</span> </span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="fl">0.1</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">'black'</span>) <span class="sc">+</span></span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_discrete</span>() <span class="sc">+</span> </span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_log10</span>() <span class="sc">+</span> </span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">width=</span><span class="fl">0.2</span>, <span class="at">outlier.size =</span> <span class="dv">0</span>, <span class="at">fill =</span> <span class="st">'white'</span>, <span class="at">color =</span> <span class="st">'black'</span>, <span class="at">coef =</span> <span class="dv">0</span>, <span class="at">outlier.color =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> cutoff.features) <span class="sc">+</span> </span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">'Feature counts'</span>) <span class="sc">+</span></span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">''</span>) <span class="sc">+</span> </span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'right'</span>)</span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a>qc.features.density <span class="sc">+</span> qc.features.violin <span class="sc">+</span> </span>
<span id="cb8-24"><a href="#cb8-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_annotation</span>(</span>
<span id="cb8-25"><a href="#cb8-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">tag_levels =</span> <span class="st">'A'</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/QCdensityGenes-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tag</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>qc.mitoRatio.density <span class="ot">=</span> sc.tagged<span class="sc">@</span>meta.data <span class="sc">%&gt;%</span> </span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">color=</span>sample, <span class="at">x=</span>mitoRatio, <span class="at">fill=</span> sample)) <span class="sc">+</span> </span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_density</span>(<span class="at">alpha =</span> <span class="fl">0.2</span>) <span class="sc">+</span> </span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_log10</span>() <span class="sc">+</span> </span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">"Cell density"</span>) <span class="sc">+</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> cutoff.mitoRatio) <span class="sc">+</span> </span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'none'</span>)</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>qc.mitoRatio.violin <span class="ot">=</span> sc.tagged<span class="sc">@</span>meta.data <span class="sc">%&gt;%</span> </span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>sample, <span class="at">y=</span>mitoRatio, <span class="at">fill=</span> sample, <span class="at">color =</span> sample)) <span class="sc">+</span> </span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_violin</span>(<span class="at">scale =</span> <span class="st">'width'</span>, <span class="at">alpha =</span> <span class="fl">0.2</span>) <span class="sc">+</span> </span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="fl">0.1</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">'black'</span>) <span class="sc">+</span></span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_discrete</span>() <span class="sc">+</span> </span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_log10</span>() <span class="sc">+</span> </span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">width=</span><span class="fl">0.2</span>, <span class="at">outlier.size =</span> <span class="dv">0</span>, <span class="at">fill =</span> <span class="st">'white'</span>, <span class="at">color =</span> <span class="st">'black'</span>, <span class="at">coef =</span> <span class="dv">0</span>, <span class="at">outlier.color =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> cutoff.mitoRatio) <span class="sc">+</span> </span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">'Feature counts'</span>) <span class="sc">+</span></span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">''</span>) <span class="sc">+</span> </span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'right'</span>)</span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a>qc.mitoRatio.density <span class="sc">+</span> qc.mitoRatio.violin <span class="sc">+</span> </span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_annotation</span>(</span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">tag_levels =</span> <span class="st">'A'</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/QCmito-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="summary" class="level2">
<h2 class="anchored" data-anchor-id="summary">Summary</h2>
<p>Summary plotting read counts vs.&nbsp;number of genes, colored by ratio of mitochondrial genes.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tag</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize the correlation between genes detected and number of UMIs and determine whether strong presence of cells with low numbers of genes/UMIs</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>qc.combined <span class="ot">=</span> sc.tagged<span class="sc">@</span>meta.data <span class="sc">%&gt;%</span> </span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>nCount_RNA, <span class="at">y=</span>nFeature_RNA)) <span class="sc">+</span> </span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">color=</span>mitoRatio), <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span> </span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>        <span class="fu">scale_colour_gradient</span>(<span class="at">low =</span> <span class="st">"gray90"</span>, <span class="at">high =</span> <span class="st">"black"</span>, <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,cutoff.mitoRatio),<span class="at">na.value =</span> <span class="st">'red'</span>) <span class="sc">+</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>      <span class="fu">stat_smooth</span>(<span class="at">method=</span>lm) <span class="sc">+</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>        <span class="fu">scale_x_log10</span>() <span class="sc">+</span> </span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>        <span class="fu">scale_y_log10</span>() <span class="sc">+</span> </span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> cutoff.reads) <span class="sc">+</span></span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> cutoff.features) <span class="sc">+</span></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>      <span class="fu">facet_wrap</span>(<span class="sc">~</span>sample)</span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>qc.combined <span class="sc">+</span> <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">'A'</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/QCnGenesnUMIsMito-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="filtering" class="level2">
<h2 class="anchored" data-anchor-id="filtering">Filtering</h2>
<p>Filtering for high-quality cells is done according to cutoffs stated above.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tag</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>sc.filt <span class="ot">=</span> sc.tagged <span class="sc">%&gt;%</span> </span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>            <span class="fu">subset</span>(nFeature_RNA <span class="sc">&gt;</span> cutoff.features) <span class="sc">%&gt;%</span> </span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>            <span class="fu">subset</span>(nCount_RNA <span class="sc">&gt;</span> cutoff.reads) <span class="sc">%&gt;%</span> </span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>            <span class="fu">subset</span>(mitoRatio <span class="sc">&lt;</span> cutoff.mitoRatio) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="dimensional-reduction" class="level1">
<h1>Dimensional reduction</h1>
<section id="normalization" class="level2">
<h2 class="anchored" data-anchor-id="normalization">Normalization</h2>
<p>Count data is scaled to 10.000 counts/cell and log2p normalized.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tag</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>sc.norm <span class="ot">=</span> sc.filt <span class="sc">%&gt;%</span> </span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>          <span class="fu">NormalizeData</span>(<span class="at">normalization.method =</span> <span class="st">"LogNormalize"</span>, <span class="at">scale.factor =</span> <span class="fl">1e4</span>) <span class="sc">%&gt;%</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>          <span class="fu">ScaleData</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="variable-genes" class="level2">
<h2 class="anchored" data-anchor-id="variable-genes">Variable genes</h2>
<p>2000 most variable genes are selected for downstream analysis.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tag</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>sc.norm <span class="ot">=</span> <span class="fu">FindVariableFeatures</span>(sc.norm)</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="fu">VariableFeaturePlot_scCustom</span>(sc.norm,<span class="at">label =</span> <span class="cn">TRUE</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/findVariableFeatures-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="pca" class="level2">
<h2 class="anchored" data-anchor-id="pca">PCA</h2>
<p>Data is normalized and PCA calculated for 2000 most variable genes. Elbow plot shows %-variance explained by each PC. There is no clear shoulder. 30 PCs will be used for downstream analysis.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tag</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>sc.norm <span class="ot">=</span> <span class="fu">RunPCA</span>(sc.norm)</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ElbowPlot</span>(sc.norm,<span class="at">ndims =</span> <span class="dv">50</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="umap" class="level2">
<h2 class="anchored" data-anchor-id="umap">UMAP</h2>
<p>For visualization, UMAP is calculated using the first 30 PC’s and default parameters (n.neighbors = 30, min.dist = 0.3, seed.use = 42, etc.). Plotted below colored by sample.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tag</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>sc.norm <span class="ot">=</span> sc.norm <span class="sc">%&gt;%</span> <span class="fu">RunUMAP</span>(<span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>,<span class="at">n.neighbors =</span> <span class="dv">30</span>,<span class="at">min.dist =</span> <span class="fl">0.3</span>, <span class="at">seed.use =</span> <span class="dv">42</span>)</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(sc.norm,<span class="at">group.by =</span> <span class="st">'sample'</span>, <span class="at">split.by =</span> <span class="st">'sample'</span>)<span class="sc">&amp;</span><span class="fu">NoAxes</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="unsupervised-clustering" class="level2">
<h2 class="anchored" data-anchor-id="unsupervised-clustering">Unsupervised clustering</h2>
<p>Nearest-neighbor and shared nearest neighbor graphs are constructed using the first 30 dimensions of PCA and standard parameters (k=20, nn.method = ‘annoy’, annoy.metric = ‘euclidean’, prune.SNN = 1/15, n.trees = 50,…) Cells are clustered by modularity optimization on the Louvain algorithm (Waltman and van Eck (2013). Standard parameters are used (resolution = 0.8).</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tag</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>sc.norm <span class="ot">=</span> sc.norm <span class="sc">%&gt;%</span> <span class="fu">FindNeighbors</span>(<span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>,<span class="at">k=</span><span class="dv">20</span>, <span class="at">nn.method  =</span> <span class="st">'annoy'</span>, <span class="at">annoy.metric =</span> <span class="st">'euclidean'</span>, <span class="at">prune.SNN  =</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">15</span>, <span class="at">n.trees =</span> <span class="dv">50</span>)</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>sc.norm <span class="ot">=</span> sc.norm <span class="sc">%&gt;%</span> <span class="fu">FindClusters</span>(<span class="at">algorithm =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 10710
Number of edges: 364263

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8374
Number of communities: 20
Elapsed time: 1 seconds</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot_scCustom</span>(sc.norm)<span class="sc">&amp;</span><span class="fu">NoAxes</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
</section>
<section id="known-markers" class="level1">
<h1>Known markers</h1>
<p>In order to assign cell types two approaches are used jointly: (1) Plotting of canonical immune cells markers (2) Calculating marker genes of individual clusters and manual search in databases with cell type annotations (proteinatlas.org, genecards.org)</p>
<section id="cd45-immune-cells" class="level2">
<h2 class="anchored" data-anchor-id="cd45-immune-cells">CD45+ immune cells</h2>
<p>Plotting Ptprc (CD45) expression shows low background from CD45- cells. Clusters 16 and 17 look like contamination.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tag</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>clPlot <span class="ot">=</span> <span class="fu">DimPlot_scCustom</span>(sc.norm)<span class="sc">&amp;</span><span class="fu">NoAxes</span>()</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Ptprc: CD45+, the cells that don't express are background </span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(sc.norm, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">'Ptprc'</span>)) <span class="sc">+</span> clPlot <span class="sc">+</span> <span class="fu">plot_layout</span>(<span class="at">ncol =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="cd14-monocytesmacrophagesmicroglia" class="level2">
<h2 class="anchored" data-anchor-id="cd14-monocytesmacrophagesmicroglia">CD14+ Monocytes/Macrophages/Microglia</h2>
<p>Plotting Cd14,C1qa,Aif1 (IBA1), Trem2 shows as signal both in macrophages and microglia</p>
<div class="cell" data-fig.asp="1">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tag</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="fu">wrap_plots</span>(<span class="fu">FeaturePlot</span>(sc.norm, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">'Cd14'</span>,<span class="st">'C1qa'</span>,<span class="st">'Aif1'</span>,<span class="st">'Trem2'</span>))<span class="sc">&amp;</span><span class="fu">NoAxes</span>())<span class="sc">+</span>clPlot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Plotting P2ry12 and Tmem 119 as markers for Microglia. Both are fairly specific to Clusters 4,5,9. C1qa and Cx3cr also have higher expression in these clusters.</p>
<div class="cell" data-fig.asp="1">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tag</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a><span class="fu">wrap_plots</span>(<span class="fu">FeaturePlot</span>(sc.norm, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">'P2ry12'</span>,<span class="st">'Tmem119'</span>,<span class="st">'C1qa'</span>,<span class="st">'Cx3cr1'</span>))<span class="sc">&amp;</span><span class="fu">NoAxes</span>())<span class="sc">+</span>clPlot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Plotting Fcgr1 (CD64), Ccr2, Cxcr4 as markers for macrophages. They are all strongest in Clusters 0,1,2,3,13</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tag</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="co">#macrophages: Fcgr1: CD64, Cx3cr1</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="fu">wrap_plots</span>(<span class="fu">FeaturePlot</span>(sc.norm, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">'Fcgr1'</span>,<span class="st">'Ccr2'</span>,<span class="st">'Cxcr4'</span>))<span class="sc">&amp;</span><span class="fu">NoAxes</span>())<span class="sc">+</span>clPlot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="cd3-t-cells" class="level2">
<h2 class="anchored" data-anchor-id="cd3-t-cells">CD3+ T-cells</h2>
<p>Plotting Cd3g, Cd3d (CD3) and Trbc1, Trbc2 (TCR) shows that clusters 7,6,12 are T-Cells. (1) Cd4 (CD4) and Cd8a (CD8) are often not well detected in scRNAseq data but can see both populations present (2) NK cells are defined by absence of CD3 and e.g.&nbsp;Klrk1 (CD314), Klrb1 (CD161), Ncr1 (NKp46) we can see a population that is CD3-,CD314+,NKp46+ in one of the T-cell clusters</p>
<div class="cell" data-fig.asp="1">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tag</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="co">#T Cells Cd3g/Cd3d: CD3</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a><span class="fu">wrap_plots</span>(<span class="fu">FeaturePlot</span>(sc.norm, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">'Cd3g'</span>,<span class="st">'Cd3d'</span>,<span class="st">'Trbc1'</span>, <span class="st">'Trbc2'</span>))<span class="sc">&amp;</span><span class="fu">NoAxes</span>())<span class="sc">+</span>clPlot </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="cd19-b-cells" class="level2">
<h2 class="anchored" data-anchor-id="cd19-b-cells">CD19+ B-cells</h2>
<p>Plotting Cd19 (CD19), Ms4a1 (CD20), Cd79a+Cd79b (CD79) shows B-cells in Cluster 18. Looks like a very small percentage, there might be some selection bias.</p>
<div class="cell" data-fig.asp="1">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tag</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a><span class="fu">wrap_plots</span>(<span class="fu">FeaturePlot</span>(sc.norm, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">'Cd19'</span>,<span class="st">'Ms4a1'</span>,<span class="st">'Cd79a'</span>,<span class="st">'Cd79b'</span>))<span class="sc">&amp;</span><span class="fu">NoAxes</span>())<span class="sc">+</span>clPlot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="conventional-dcs" class="level2">
<h2 class="anchored" data-anchor-id="conventional-dcs">Conventional DCs</h2>
<p>Plotting Itgax (CD11c), H2-Aa, H2-ab1, H2-D1 (MHCII) as markers for conventional dendritic cells (DC). Both are not very specific, strongest in Cluster 10. Additionally, Xcr1 and Clec9a (CD370) are supposed to be markers for DC presenting to CD8+ T-cells which are also concentrated in Cluster 10.</p>
<div class="cell" data-fig.asp="1">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tag</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="fu">wrap_plots</span>(<span class="fu">FeaturePlot</span>(sc.norm, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">'Itgax'</span>,<span class="st">'H2-Aa'</span>,<span class="st">'Xcr1'</span>,<span class="st">'Clec9a'</span>))<span class="sc">&amp;</span><span class="fu">NoAxes</span>())<span class="sc">+</span>clPlot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="siglec-h-plasmacytoid-dcs" class="level2">
<h2 class="anchored" data-anchor-id="siglec-h-plasmacytoid-dcs">Siglec-H+ plasmacytoid DCs</h2>
<p>Plotting Siglech (Siglec-H), which also marks Microglia and Bst2 (CD317) as Markers of Plasmacytoid DCs. Both are strongest in Cluster 14.</p>
<div class="cell" data-fig.asp="1">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tag</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="fu">wrap_plots</span>(<span class="fu">FeaturePlot</span>(sc.norm, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">'Siglech'</span>,<span class="st">'Bst2'</span>),<span class="at">ncol =</span> <span class="dv">1</span>) <span class="sc">&amp;</span> <span class="fu">NoAxes</span>()) <span class="sc">+</span> clPlot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="granulocytes" class="level2">
<h2 class="anchored" data-anchor-id="granulocytes">Granulocytes</h2>
<p>Expression of Ngp (Neutrophilic granule protein), S100a8 and Retnlg identfy Cluster 15 as granulocytes –&gt; these are neutrophils that are important for EAE additional markers: Cxcr2: main chemokine receptor for neutrophils, Ly6g</p>
<div class="cell" data-fig.asp="1">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tag</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="fu">wrap_plots</span>(<span class="fu">FeaturePlot</span>(sc.norm, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">'S100a8'</span>,<span class="st">'Ngp'</span>,<span class="st">'Retnlg'</span>))<span class="sc">&amp;</span><span class="fu">NoAxes</span>()<span class="sc">&amp;</span><span class="fu">NoAxes</span>())<span class="sc">+</span>clPlot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="identifying-remaining-clusters" class="level2">
<h2 class="anchored" data-anchor-id="identifying-remaining-clusters">Identifying remaining clusters</h2>
<p>Clusters 16, 11 and 17 still remain unclear. Checking by calculating unqiue clustermarkers:</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tag</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="fu">FindMarkers</span>(sc.norm, <span class="at">group.by =</span> <span class="st">'seurat_clusters'</span>,<span class="at">ident.1 =</span> <span class="dv">17</span>, <span class="at">logfc.threshold =</span> <span class="dv">2</span>, <span class="at">min.diff.pct =</span> <span class="fl">0.7</span>, <span class="at">only.pos =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>                p_val avg_log2FC pct.1 pct.2     p_val_adj
Tm4sf1   0.000000e+00   2.805910 0.707 0.002  0.000000e+00
Id3      0.000000e+00   2.596465 0.747 0.016  0.000000e+00
Sparcl1  0.000000e+00   3.010900 0.813 0.003  0.000000e+00
Ptn      0.000000e+00   3.544829 0.787 0.008  0.000000e+00
Nedd4    0.000000e+00   2.546426 0.773 0.014  0.000000e+00
Igfbp7  5.368829e-211   4.577987 0.813 0.047 1.667183e-206
Tsc22d1 6.104530e-184   2.607984 0.813 0.052 1.895640e-179
Sparc    1.052539e-68   2.639328 0.987 0.220  3.268449e-64</code></pre>
</div>
</div>
<p>Cluster 17 seems to be oligendrocytes (Mag, Klk6, Ptgds)</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tag</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="fu">FindMarkers</span>(sc.norm, <span class="at">group.by =</span> <span class="st">'seurat_clusters'</span>,<span class="at">ident.1 =</span> <span class="dv">16</span>, <span class="at">logfc.threshold =</span> <span class="dv">2</span>, <span class="at">min.diff.pct =</span> <span class="fl">0.7</span>, <span class="at">only.pos =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>              p_val avg_log2FC pct.1 pct.2   p_val_adj
S100a9 0.000000e+00   8.055196 0.901 0.013 0.00000e+00
S100a8 9.247286e-81   7.028866 0.938 0.228 2.87156e-76</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>sc.norm <span class="sc">%&gt;%</span> <span class="fu">FeaturePlot</span>(<span class="st">'Cldn5'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Tm4sf1 points to endothelial cells for cluster 16, which can be confirmed by looking at Cldn5</p>
<p>One cluster remains unannotated. By calculating marker genes we find Fscn1 and Ccr7 among the top markers which are annotated in literature as highly specific markers for activation of dendritic cells.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tag</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>sc.norm <span class="sc">%&gt;%</span> <span class="fu">FindMarkers</span>(<span class="at">ident.1 =</span> <span class="st">'11'</span>,<span class="at">logfc.threshold =</span> <span class="fl">0.5</span>, <span class="at">only.pos =</span> <span class="cn">TRUE</span>,<span class="at">min.diff.pct =</span> <span class="fl">0.7</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>                  p_val avg_log2FC pct.1 pct.2     p_val_adj
Fscn1      0.000000e+00   4.467155 0.905 0.058  0.000000e+00
Ramp3      0.000000e+00   2.848859 0.709 0.008  0.000000e+00
Ccr7       0.000000e+00   4.102384 0.921 0.007  0.000000e+00
Serpinb6b  0.000000e+00   3.729916 0.852 0.034  0.000000e+00
Traf1     1.047943e-240   2.608671 0.820 0.100 3.254177e-236
Cytip     6.252751e-192   2.987493 0.931 0.183 1.941667e-187
Tmem123   4.074869e-170   3.310465 0.942 0.241 1.265369e-165</code></pre>
</div>
</div>
</section>
<section id="assigning-preliminary-cell-type-labels." class="level2">
<h2 class="anchored" data-anchor-id="assigning-preliminary-cell-type-labels.">Assigning preliminary cell type labels.</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tag</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="co">#assigning original cell type labels based on old seurat clusters</span></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>sc.norm<span class="sc">$</span>clusters <span class="ot">=</span> sc.norm<span class="sc">$</span>seurat_clusters</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(sc.norm) <span class="ot">=</span> sc.norm<span class="sc">$</span>clusters</span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>sc.norm <span class="ot">=</span> <span class="fu">RenameIdents</span>(sc.norm,</span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a><span class="st">'0'</span> <span class="ot">=</span> <span class="st">'Monocytes/Macrophages'</span>,</span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a><span class="st">'1'</span> <span class="ot">=</span> <span class="st">'Monocytes/Macrophages'</span>,</span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a><span class="st">'2'</span> <span class="ot">=</span> <span class="st">'Monocytes/Macrophages'</span>,</span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a><span class="st">'3'</span> <span class="ot">=</span> <span class="st">'Monocytes/Macrophages'</span>,</span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a><span class="st">'13'</span> <span class="ot">=</span> <span class="st">'Monocytes/Macrophages'</span>,</span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a><span class="st">'4'</span> <span class="ot">=</span> <span class="st">'Microglia'</span>,</span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a><span class="st">'5'</span> <span class="ot">=</span> <span class="st">'Microglia'</span>,</span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a><span class="st">'9'</span> <span class="ot">=</span> <span class="st">'Microglia'</span>,</span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a><span class="st">'6'</span> <span class="ot">=</span> <span class="st">'T-Cells'</span>,</span>
<span id="cb35-15"><a href="#cb35-15" aria-hidden="true" tabindex="-1"></a><span class="st">'7'</span> <span class="ot">=</span> <span class="st">'T-Cells'</span>,</span>
<span id="cb35-16"><a href="#cb35-16" aria-hidden="true" tabindex="-1"></a><span class="st">'12'</span> <span class="ot">=</span> <span class="st">'T-Cells'</span>,</span>
<span id="cb35-17"><a href="#cb35-17" aria-hidden="true" tabindex="-1"></a><span class="st">'8'</span> <span class="ot">=</span> <span class="st">'Cycling cells'</span>,</span>
<span id="cb35-18"><a href="#cb35-18" aria-hidden="true" tabindex="-1"></a><span class="st">'10'</span> <span class="ot">=</span> <span class="st">'Conventional DC'</span>,</span>
<span id="cb35-19"><a href="#cb35-19" aria-hidden="true" tabindex="-1"></a><span class="st">'11'</span> <span class="ot">=</span> <span class="st">'Activated DC'</span>,</span>
<span id="cb35-20"><a href="#cb35-20" aria-hidden="true" tabindex="-1"></a><span class="st">'14'</span> <span class="ot">=</span> <span class="st">'Plasmacytoid DC'</span>,</span>
<span id="cb35-21"><a href="#cb35-21" aria-hidden="true" tabindex="-1"></a><span class="st">'15'</span> <span class="ot">=</span> <span class="st">'Granulocytes'</span>,</span>
<span id="cb35-22"><a href="#cb35-22" aria-hidden="true" tabindex="-1"></a><span class="st">'16'</span> <span class="ot">=</span> <span class="st">'Endothelial cells'</span>,</span>
<span id="cb35-23"><a href="#cb35-23" aria-hidden="true" tabindex="-1"></a><span class="st">'17'</span> <span class="ot">=</span> <span class="st">'Glia cells'</span>,</span>
<span id="cb35-24"><a href="#cb35-24" aria-hidden="true" tabindex="-1"></a><span class="st">'18'</span> <span class="ot">=</span> <span class="st">'B-Cells'</span>)</span>
<span id="cb35-25"><a href="#cb35-25" aria-hidden="true" tabindex="-1"></a>sc.norm<span class="sc">$</span>celltypes <span class="ot">=</span> <span class="fu">Idents</span>(sc.norm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="cell-cycle" class="level2">
<h2 class="anchored" data-anchor-id="cell-cycle">Cell cycle</h2>
<p>Cell cycle phase scoring (using an enrichment approach adapted from Tirosh et al.&nbsp;2016) shows actively dividing cells only in Cluster 8 which is recapitulated by expression of canonical markers for proliferating cells Top2a and Mki67.This cluster shows markers for both T-cells and microglia so we have to regress out cell cycle phase to put each cell type in their corresponding clusters.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>cell_cycle_genes <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">'Mus_musculus_CC.csv'</span>)</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Download the appropriate Ensembldb database</span></span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>annotations <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">'geneInfoEns105.csv'</span>)</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Get gene names for Ensembl IDs for each gene</span></span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>cell_cycle_markers <span class="ot">=</span> <span class="fu">left_join</span>(cell_cycle_genes, annotations, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"geneID"</span> <span class="ot">=</span> <span class="st">"gene_id"</span>))</span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Acquire the S phase genes</span></span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a>s_genes <span class="ot">=</span> cell_cycle_markers <span class="sc">%&gt;%</span></span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(phase <span class="sc">==</span> <span class="st">"S"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a>        <span class="fu">pull</span>(<span class="st">"gene_name"</span>)</span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb36-13"><a href="#cb36-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Acquire the G2M phase genes        </span></span>
<span id="cb36-14"><a href="#cb36-14" aria-hidden="true" tabindex="-1"></a>g2m_genes <span class="ot">=</span> cell_cycle_markers <span class="sc">%&gt;%</span></span>
<span id="cb36-15"><a href="#cb36-15" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(phase <span class="sc">==</span> <span class="st">"G2/M"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-16"><a href="#cb36-16" aria-hidden="true" tabindex="-1"></a>        <span class="fu">pull</span>(<span class="st">"gene_name"</span>)</span>
<span id="cb36-17"><a href="#cb36-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform cell cycle scoring</span></span>
<span id="cb36-18"><a href="#cb36-18" aria-hidden="true" tabindex="-1"></a>sc.norm <span class="ot">=</span> <span class="fu">CellCycleScoring</span>(sc.norm,</span>
<span id="cb36-19"><a href="#cb36-19" aria-hidden="true" tabindex="-1"></a>                           <span class="at">g2m.features =</span> g2m_genes,</span>
<span id="cb36-20"><a href="#cb36-20" aria-hidden="true" tabindex="-1"></a>                           <span class="at">s.features =</span> s_genes)</span>
<span id="cb36-21"><a href="#cb36-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-22"><a href="#cb36-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-23"><a href="#cb36-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize the UMAP, grouping by cell cycle phase</span></span>
<span id="cb36-24"><a href="#cb36-24" aria-hidden="true" tabindex="-1"></a><span class="fu">wrap_plots</span>(<span class="fu">DimPlot</span>(sc.norm,<span class="at">group.by=</span> <span class="st">"Phase"</span>, <span class="at">split.by =</span> <span class="st">'Phase'</span>,<span class="at">ncol =</span> <span class="dv">1</span>)<span class="sc">&amp;</span><span class="fu">NoAxes</span>()) <span class="sc">+</span> <span class="fu">wrap_plots</span>( <span class="fu">FeaturePlot</span>(sc.norm,<span class="at">features =</span> <span class="fu">c</span>(<span class="st">'Top2a'</span>,<span class="st">'Mki67'</span>), <span class="at">ncol =</span> <span class="dv">1</span>)<span class="sc">&amp;</span><span class="fu">NoAxes</span>()) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>scCC <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(sc.norm, <span class="at">vars.to.regress =</span> <span class="fu">c</span>(<span class="st">"S.Score"</span>, <span class="st">"G2M.Score"</span>), <span class="at">features =</span> <span class="fu">rownames</span>(sc.norm))</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>scCC <span class="ot">=</span> scCC <span class="sc">%&gt;%</span> </span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>        RunPCA <span class="sc">%&gt;%</span> </span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>        <span class="fu">RunUMAP</span>(<span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>,<span class="at">n.neighbors =</span> <span class="dv">30</span>,<span class="at">min.dist =</span> <span class="fl">0.3</span>, <span class="at">seed.use =</span> <span class="dv">42</span>)</span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a><span class="fu">SaveH5Seurat</span>(scCC,<span class="fu">paste0</span>(outdir,<span class="st">'scCC.h5seurat'</span>,<span class="at">overwrite =</span> <span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Now cycling cells are closer to the microglia and t-cell clusters</p>
<div class="cell" data-fig.asp="0.5">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tag</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="co">#Here we can see that 'cycling cells' cluster is now split mostly between Microglia and T-Cells </span></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>scCC <span class="sc">%&gt;%</span> <span class="fu">DimPlot_scCustom</span>(<span class="at">label=</span><span class="cn">TRUE</span>,<span class="at">group.by =</span> <span class="st">'celltypes'</span>) <span class="sc">+</span> </span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>scCC <span class="sc">%&gt;%</span> <span class="fu">FeaturePlot</span>(<span class="fu">c</span>(<span class="st">'Top2a'</span>,<span class="st">'Mki67'</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid" width="1920"></p>
</div>
</div>
<p>## Doublet &amp; contamination removal</p>
<p>In order to improve the clustering we will also remove doublet cells using scDoubletFinder and filter out any clusters that are not immune cells. As scDoubletFinder relies on randomization, the clustering results downstream of that will not be 100% reproducible via this notebook. Total number of clusters and assignment of cells to (unsupervised) clusters will vary slightly. Therefore we provide the code we used and will then load a list containing the filtered cell barcodes as they were output when we ran the function at the end of the snippet in order to generate exactly the same figures as presented in the paper.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tag</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>sce <span class="ot">=</span> <span class="fu">as.SingleCellExperiment</span>(scCC)</span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>sce.dbl <span class="ot">=</span> <span class="fu">scDblFinder</span>(sce)</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>(<span class="fu">plotUMAP</span>(sce,<span class="at">colour_by =</span> <span class="st">'celltypes'</span>) <span class="sc">+</span> </span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a> <span class="fu">plotUMAP</span>(sce.dbl,<span class="at">colour_by =</span> <span class="st">'scDblFinder.score'</span>)) <span class="sc">/</span> </span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>(<span class="fu">plotUMAP</span>(sce.dbl,<span class="at">colour_by =</span> <span class="st">'nCount_RNA'</span>) <span class="sc">+</span> </span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a> <span class="fu">plotUMAP</span>(sce.dbl,<span class="at">colour_by =</span> <span class="st">'nFeature_RNA'</span>)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tag</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a><span class="co">#scdblfinder also predicts which cells are singlets absolutly </span></span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plotUMAP</span>(sce.dbl,<span class="at">colour_by =</span> <span class="st">'scDblFinder.class'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co">#and we can use this information to subset the seurat object </span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>sce.noDbl <span class="ot">=</span> sce.dbl[,sce.dbl<span class="sc">$</span>scDblFinder.class <span class="sc">==</span> <span class="st">'singlet'</span>]</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>scCC.noDbl <span class="ot">=</span> scCC[,sce.dbl<span class="sc">$</span>scDblFinder.class <span class="sc">==</span> <span class="st">'singlet'</span>]</span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a><span class="co">#after throwing out doublets recompute dimensional reductions and clustering </span></span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a>scCC.noDblRE <span class="ot">=</span> scCC.noDbl <span class="sc">%&gt;%</span> </span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a>                RunPCA <span class="sc">%&gt;%</span>  </span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a>                <span class="fu">RunUMAP</span>(<span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>,<span class="at">n.neighbors =</span> <span class="dv">30</span>,<span class="at">min.dist =</span> <span class="fl">0.3</span>, <span class="at">seed.use =</span> <span class="dv">42</span>) <span class="sc">%&gt;%</span>  </span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a>                <span class="fu">FindNeighbors</span>(<span class="at">dims =</span>  <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb41-11"><a href="#cb41-11" aria-hidden="true" tabindex="-1"></a>                <span class="fu">FindClusters</span>(<span class="at">resolution =</span> <span class="fl">1.1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb41-12"><a href="#cb41-12" aria-hidden="true" tabindex="-1"></a>                <span class="fu">BuildClusterTree</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 9095
Number of edges: 309425

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.7922
Number of communities: 21
Elapsed time: 1 seconds</code></pre>
</div>
</div>
<div class="cell" data-fig.asp="0.5">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tag</span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>scCC.noDblRE.clean  <span class="ot">=</span> scCC.noDblRE[,scCC.noDblRE<span class="sc">$</span>seurat_clusters <span class="sc">%!in%</span> <span class="fu">c</span>(<span class="dv">19</span>,<span class="dv">20</span>)]</span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a><span class="co">#scCC.noDblRE.clean %&gt;% colnames %&gt;% </span></span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a><span class="co">#                       as.data.frame() %&gt;% </span></span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a><span class="co">#                       write_csv(file = 'scCC.noDblRE.clean_barcodes.csv')</span></span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a>filtered_bc <span class="ot">=</span> <span class="fu">read_csv</span>(<span class="at">file =</span> <span class="st">'scCC.noDblRE.clean_barcodes.csv'</span>)<span class="sc">$</span>.</span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a>scCC.noDblRE.clean <span class="ot">=</span> scCC[,filtered_bc]</span>
<span id="cb43-9"><a href="#cb43-9" aria-hidden="true" tabindex="-1"></a>scCC.noDblRE.clean <span class="ot">=</span> scCC.noDblRE.clean <span class="sc">%&gt;%</span> </span>
<span id="cb43-10"><a href="#cb43-10" aria-hidden="true" tabindex="-1"></a>                      RunPCA <span class="sc">%&gt;%</span>  </span>
<span id="cb43-11"><a href="#cb43-11" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">RunUMAP</span>(<span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>,<span class="at">n.neighbors =</span> <span class="dv">30</span>,<span class="at">min.dist =</span> <span class="fl">0.3</span>, <span class="at">seed.use =</span> <span class="dv">42</span>) <span class="sc">%&gt;%</span>  </span>
<span id="cb43-12"><a href="#cb43-12" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">FindNeighbors</span>(<span class="at">dims =</span>  <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb43-13"><a href="#cb43-13" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">FindClusters</span>(<span class="at">resolution =</span> <span class="fl">0.8</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb43-14"><a href="#cb43-14" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">BuildClusterTree</span>() </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 8954
Number of edges: 307147

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8154
Number of communities: 16
Elapsed time: 1 seconds</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>scCC.noDblRE.clean <span class="sc">%&gt;%</span> <span class="fu">subset</span>(celltypes <span class="sc">%!in%</span> <span class="fu">levels</span>(scCC.noDblRE.clean<span class="sc">$</span>celltypes)[<span class="dv">9</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>An object of class Seurat 
31053 features across 8953 samples within 1 assay 
Active assay: RNA (31053 features, 2000 variable features)
 2 dimensional reductions calculated: pca, umap</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>scCC.noDblRE.clean <span class="sc">%&gt;%</span> <span class="fu">DimPlot_scCustom</span>(<span class="at">label =</span> <span class="cn">TRUE</span>) <span class="sc">+</span> scCC.noDblRE.clean <span class="sc">%&gt;%</span> <span class="fu">DimPlot_scCustom</span>(<span class="at">group.by =</span> <span class="st">'celltypes'</span>, <span class="at">label =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid" width="1920"></p>
</div>
</div>
<p>Now we assign final cell type labels.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tag</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>scCC.noDblRE.clean<span class="sc">$</span>clusters <span class="ot">=</span> scCC.noDblRE.clean<span class="sc">$</span>seurat_clusters</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(scCC.noDblRE.clean) <span class="ot">=</span> scCC.noDblRE.clean<span class="sc">$</span>clusters</span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>scCC.noDblRE.clean <span class="ot">=</span> <span class="fu">RenameIdents</span>(scCC.noDblRE.clean,</span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a><span class="st">'0'</span> <span class="ot">=</span> <span class="st">'Macrophages'</span>,</span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a><span class="st">'1'</span> <span class="ot">=</span> <span class="st">'Macrophages'</span>,</span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a><span class="st">'2'</span> <span class="ot">=</span> <span class="st">'Macrophages'</span>,</span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a><span class="st">'3'</span> <span class="ot">=</span> <span class="st">'Macrophages'</span>,</span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a><span class="st">'4'</span> <span class="ot">=</span> <span class="st">'Microglia'</span>,</span>
<span id="cb48-12"><a href="#cb48-12" aria-hidden="true" tabindex="-1"></a><span class="st">'5'</span> <span class="ot">=</span> <span class="st">'Microglia'</span>,</span>
<span id="cb48-13"><a href="#cb48-13" aria-hidden="true" tabindex="-1"></a><span class="st">'12'</span> <span class="ot">=</span> <span class="st">'Microglia'</span>,</span>
<span id="cb48-14"><a href="#cb48-14" aria-hidden="true" tabindex="-1"></a><span class="st">'14'</span> <span class="ot">=</span> <span class="st">'Microglia'</span>,</span>
<span id="cb48-15"><a href="#cb48-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-16"><a href="#cb48-16" aria-hidden="true" tabindex="-1"></a><span class="st">'6'</span> <span class="ot">=</span> <span class="st">'T-Cells'</span>,</span>
<span id="cb48-17"><a href="#cb48-17" aria-hidden="true" tabindex="-1"></a><span class="st">'8'</span> <span class="ot">=</span> <span class="st">'T-Cells'</span>,</span>
<span id="cb48-18"><a href="#cb48-18" aria-hidden="true" tabindex="-1"></a><span class="st">'11'</span> <span class="ot">=</span> <span class="st">'T-Cells'</span>,</span>
<span id="cb48-19"><a href="#cb48-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-20"><a href="#cb48-20" aria-hidden="true" tabindex="-1"></a><span class="st">'7'</span> <span class="ot">=</span> <span class="st">'Conventional DC'</span>,</span>
<span id="cb48-21"><a href="#cb48-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-22"><a href="#cb48-22" aria-hidden="true" tabindex="-1"></a><span class="st">'9'</span> <span class="ot">=</span> <span class="st">'Activated DC'</span>,</span>
<span id="cb48-23"><a href="#cb48-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-24"><a href="#cb48-24" aria-hidden="true" tabindex="-1"></a><span class="st">'10'</span> <span class="ot">=</span> <span class="st">'Plasmacytoid DC'</span>,</span>
<span id="cb48-25"><a href="#cb48-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-26"><a href="#cb48-26" aria-hidden="true" tabindex="-1"></a><span class="st">'13'</span> <span class="ot">=</span> <span class="st">'Granulocytes'</span>,</span>
<span id="cb48-27"><a href="#cb48-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-28"><a href="#cb48-28" aria-hidden="true" tabindex="-1"></a><span class="st">'15'</span> <span class="ot">=</span> <span class="st">'B-Cells'</span>)</span>
<span id="cb48-29"><a href="#cb48-29" aria-hidden="true" tabindex="-1"></a>scCC.noDblRE.clean<span class="sc">$</span>celltypesNew <span class="ot">=</span> <span class="fu">Idents</span>(scCC.noDblRE.clean)</span>
<span id="cb48-30"><a href="#cb48-30" aria-hidden="true" tabindex="-1"></a>scCC.noDblRE.clean <span class="sc">%&gt;%</span> <span class="fu">DimPlot_scCustom</span>(<span class="at">group.by =</span> <span class="st">'celltypesNew'</span>,<span class="at">label =</span> <span class="cn">TRUE</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/unnamed-chunk-25-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tag</span></span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a><span class="fu">SaveH5Seurat</span>(scCC.noDblRE.clean, <span class="at">file =</span> <span class="fu">paste0</span>(outdir,<span class="st">'/MQ_EAE_processed.h5seurat'</span>),<span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="subclustering-analysis" class="level1">
<h1>Subclustering analysis</h1>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>  sc.IC <span class="ot">=</span> <span class="fu">LoadH5Seurat</span>(<span class="at">file =</span> <span class="fu">paste0</span>(indir,<span class="st">'GSE222063_MQ_EAE_processed.h5seurat'</span>))</span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>  sc.mg <span class="ot">=</span> sc.IC <span class="sc">%&gt;%</span> <span class="fu">subset</span>(celltypesNew <span class="sc">==</span> <span class="st">'Microglia'</span>)</span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a>  sc.mac <span class="ot">=</span> sc.IC <span class="sc">%&gt;%</span> <span class="fu">subset</span>(celltypesNew <span class="sc">==</span> <span class="st">'Macrophages'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="macrophages" class="level2">
<h2 class="anchored" data-anchor-id="macrophages">Macrophages</h2>
<section id="integration-reanalysis-of-macrophages" class="level3">
<h3 class="anchored" data-anchor-id="integration-reanalysis-of-macrophages">Integration &amp; reanalysis of Macrophages</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>sc.split <span class="ot">=</span> <span class="fu">SplitObject</span>(sc.mac,<span class="at">split.by =</span> <span class="st">'sample'</span>)</span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a><span class="co"># normalize and identify variable features for each dataset independently</span></span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a>sc.split <span class="ot">&lt;-</span> <span class="fu">lapply</span>(sc.split,<span class="cf">function</span>(x) {</span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(x)</span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(x, <span class="at">selection.method =</span> <span class="st">"vst"</span>, <span class="at">nfeatures =</span> <span class="dv">2000</span>)</span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb51-8"><a href="#cb51-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-9"><a href="#cb51-9" aria-hidden="true" tabindex="-1"></a><span class="co"># select features that are repeatedly variable across datasets for integration</span></span>
<span id="cb51-10"><a href="#cb51-10" aria-hidden="true" tabindex="-1"></a>features <span class="ot">&lt;-</span> <span class="fu">SelectIntegrationFeatures</span>(sc.split)</span>
<span id="cb51-11"><a href="#cb51-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-12"><a href="#cb51-12" aria-hidden="true" tabindex="-1"></a><span class="co"># find features for integration</span></span>
<span id="cb51-13"><a href="#cb51-13" aria-hidden="true" tabindex="-1"></a>mac.anchors <span class="ot">&lt;-</span> <span class="fu">FindIntegrationAnchors</span>(sc.split, <span class="at">anchor.features =</span> features)</span>
<span id="cb51-14"><a href="#cb51-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-15"><a href="#cb51-15" aria-hidden="true" tabindex="-1"></a><span class="co"># create integrated assay</span></span>
<span id="cb51-16"><a href="#cb51-16" aria-hidden="true" tabindex="-1"></a>mac.combined <span class="ot">&lt;-</span> <span class="fu">IntegrateData</span>(<span class="at">anchorset =</span> mac.anchors)</span>
<span id="cb51-17"><a href="#cb51-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-18"><a href="#cb51-18" aria-hidden="true" tabindex="-1"></a><span class="co"># reanalyse integrated data </span></span>
<span id="cb51-19"><a href="#cb51-19" aria-hidden="true" tabindex="-1"></a><span class="fu">DefaultAssay</span>(mac.combined) <span class="ot">=</span> <span class="st">'integrated'</span></span>
<span id="cb51-20"><a href="#cb51-20" aria-hidden="true" tabindex="-1"></a>mac.combined <span class="ot">=</span> mac.combined <span class="sc">%&gt;%</span>       <span class="fu">ScaleData</span>() <span class="sc">%&gt;%</span></span>
<span id="cb51-21"><a href="#cb51-21" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">RunPCA</span>(<span class="at">npcs =</span> <span class="dv">30</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb51-22"><a href="#cb51-22" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">RunUMAP</span>(<span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb51-23"><a href="#cb51-23" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">FindNeighbors</span>(<span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb51-24"><a href="#cb51-24" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">FindClusters</span>(<span class="at">resolution =</span> <span class="fl">0.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 6102
Number of edges: 202931

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.7706
Number of communities: 6
Elapsed time: 0 seconds</code></pre>
</div>
</div>
</section>
<section id="identify-contaminating-cell-types" class="level3">
<h3 class="anchored" data-anchor-id="identify-contaminating-cell-types">Identify contaminating cell types</h3>
<div class="cell" data-fig.asp="1">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>genes_celltypes <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_b =</span> <span class="fu">c</span>(<span class="st">'Cd19'</span>,<span class="st">'Ms4a1'</span>),</span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_gran =</span> <span class="fu">c</span>(<span class="st">'S100a8'</span>,<span class="st">'Retnlg'</span>),</span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_pdc =</span> <span class="fu">c</span>(<span class="st">'Siglech'</span>,<span class="st">'Bst2'</span>),</span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_cdcAct =</span> <span class="fu">c</span>(<span class="st">'Fscn1'</span>,<span class="st">'Ccr7'</span>),</span>
<span id="cb53-6"><a href="#cb53-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_cdc =</span> <span class="fu">c</span>(<span class="st">'Itgax'</span>,<span class="st">'Btla'</span>),</span>
<span id="cb53-7"><a href="#cb53-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_t =</span> <span class="fu">c</span>(<span class="st">'Cd3g'</span>,<span class="st">'Trbc1'</span>),</span>
<span id="cb53-8"><a href="#cb53-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_mg =</span> <span class="fu">c</span>(<span class="st">'Tmem119'</span>,<span class="st">'Cx3cr1'</span>),</span>
<span id="cb53-9"><a href="#cb53-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">gene_mac =</span> <span class="fu">c</span>(<span class="st">'Arg1'</span>,<span class="st">'Cd68'</span>)</span>
<span id="cb53-10"><a href="#cb53-10" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb53-11"><a href="#cb53-11" aria-hidden="true" tabindex="-1"></a>mac.combined <span class="sc">%&gt;%</span> <span class="fu">DimPlot_scCustom</span>(<span class="at">label =</span> <span class="cn">TRUE</span>) <span class="sc">/</span> <span class="fu">wrap_plots</span>(</span>
<span id="cb53-12"><a href="#cb53-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lapply</span>(genes_celltypes,<span class="cf">function</span>(x){</span>
<span id="cb53-13"><a href="#cb53-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">FeaturePlot_scCustom</span>(mac.combined,x,<span class="at">colors_use =</span> viridis_light_high) <span class="sc">+</span> </span>
<span id="cb53-14"><a href="#cb53-14" aria-hidden="true" tabindex="-1"></a>      <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'right'</span>,</span>
<span id="cb53-15"><a href="#cb53-15" aria-hidden="true" tabindex="-1"></a>            <span class="at">legend.box =</span> <span class="st">'horizontal'</span>,</span>
<span id="cb53-16"><a href="#cb53-16" aria-hidden="true" tabindex="-1"></a>            <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">4</span>))</span>
<span id="cb53-17"><a href="#cb53-17" aria-hidden="true" tabindex="-1"></a>  })) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/unnamed-chunk-28-1.png" class="img-fluid" width="1440"></p>
</div>
</div>
<p>Based on marker genes for the individual cell types identified in our dataset no single cluster seems to contain a contamination.</p>
</section>
<section id="identification-of-specific-macrophage-cell-states" class="level3">
<h3 class="anchored" data-anchor-id="identification-of-specific-macrophage-cell-states">Identification of specific macrophage cell states</h3>
<p>Cell states were analyzed based on discussion and marker gene expression reported in Giladi et al.&nbsp;2021. Here they state, that Ly6c+ monocytes (human: CD14hi-CD16lo) carry chemokine receptors that allow tissue infiltration and subsequent generation of different monocyte-derived cells. After infiltration they gain expression of MHCII-related genes. Monocytes expressing Ccr2 were identified as main drivers of EAE pathogenesis.</p>
<p>In their single-cell analysis they could identify a more fine-grained classification of different cell states:</p>
<ul>
<li>One cluster showed expression of interferon responsive genes and labeled Ifi2+</li>
<li>Two subsets expressed Arg1, Apoc2 and C1qb and were designated as Arg1+ I and</li>
<li>A cluster was characterized by expression of Nos2, Gpnmb, Arg1 and Fabp5 and designated Nos2+</li>
<li>One population that expressed inflammatory genes such as Saa3, Plac8 and Gbp2 was called Saa3+</li>
<li>A cluster expressing Cxcl9, Cxcl10 and Il1b was designated as Cxcl10+</li>
</ul>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>genes_mac <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_general =</span> <span class="fu">c</span>(<span class="st">'Ly6c2'</span>, <span class="st">'Ccr2'</span>,<span class="st">'H2-Ab1'</span>,<span class="st">'H2-D1'</span>,<span class="st">'H2-Eb1'</span>,<span class="st">'Top2a'</span>,<span class="st">'Mki67'</span>),</span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_arg1 =</span> <span class="fu">c</span>(<span class="st">'Arg1'</span>,<span class="st">'Ecm1'</span>,<span class="st">'Tmem37'</span>,<span class="st">'Apoc2'</span>,<span class="st">'C1qb'</span>,<span class="st">'Itgb5'</span>,<span class="st">'Itga6'</span>,<span class="st">'Anxa3'</span>,<span class="st">'Ptgs1'</span>,<span class="st">'C3ar1'</span>),</span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_arg2 =</span> <span class="fu">c</span>(<span class="st">'Cd14'</span>,<span class="st">'Arg1'</span>,<span class="st">'Thbs1'</span>,<span class="st">'Mmp19'</span>,<span class="st">'Klf2'</span>,<span class="st">'Ecm1'</span>,<span class="st">'Lmna'</span>),</span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_nos2 =</span> <span class="fu">c</span>(<span class="st">'Gpnmb'</span>,<span class="st">'Fabp5'</span>,<span class="st">'Nos2'</span>,<span class="st">'Slc7a2'</span>,<span class="st">'Slc7a11'</span>,<span class="st">'Fth1'</span>,<span class="st">'Cstb'</span>),</span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_ifit1 =</span> <span class="fu">c</span>(<span class="st">'Ifit1'</span>,<span class="st">'Ifit2'</span>,<span class="st">'Ifit3'</span>, <span class="st">'Usp18'</span>,<span class="st">'Irf7'</span>),</span>
<span id="cb54-7"><a href="#cb54-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_saa3 =</span> <span class="fu">c</span>(<span class="st">'Saa3'</span>,<span class="st">'Saal1'</span>,<span class="st">'Tgm2'</span>,<span class="st">'Gbp2'</span>,<span class="st">'Upp1'</span>,<span class="st">'Plac8'</span>,<span class="st">'Clec5a'</span>),</span>
<span id="cb54-8"><a href="#cb54-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_cxcl10 =</span> <span class="fu">c</span>(<span class="st">'Cxcl10'</span>,<span class="st">'Cxcl9'</span>,<span class="st">'Cxcl11'</span>,<span class="st">'Socs1'</span>,<span class="st">'Gbp2'</span>,<span class="st">'Tgm2'</span>,<span class="st">'Nod1'</span>,<span class="st">'Upp1'</span>,<span class="st">'Il1b'</span>,<span class="st">'Smox'</span>))</span>
<span id="cb54-9"><a href="#cb54-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-10"><a href="#cb54-10" aria-hidden="true" tabindex="-1"></a>mac_states_plots <span class="ot">=</span> <span class="fu">lapply</span>(genes_mac,<span class="cf">function</span>(x){</span>
<span id="cb54-11"><a href="#cb54-11" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">wrap_plots</span>(<span class="at">design =</span> <span class="st">'ABBB</span></span>
<span id="cb54-12"><a href="#cb54-12" aria-hidden="true" tabindex="-1"></a><span class="st">                                           #BBB'</span>,</span>
<span id="cb54-13"><a href="#cb54-13" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">DimPlot_scCustom</span>(<span class="at">seurat_object =</span> mac.combined,</span>
<span id="cb54-14"><a href="#cb54-14" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">group.by =</span> <span class="st">'seurat_clusters'</span>,</span>
<span id="cb54-15"><a href="#cb54-15" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">label =</span> <span class="cn">TRUE</span>,</span>
<span id="cb54-16"><a href="#cb54-16" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">label.size =</span> <span class="dv">10</span>), </span>
<span id="cb54-17"><a href="#cb54-17" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">FeaturePlot_scCustom</span>(<span class="at">seurat_object =</span> mac.combined,</span>
<span id="cb54-18"><a href="#cb54-18" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">features =</span> x,</span>
<span id="cb54-19"><a href="#cb54-19" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">colors_use =</span> viridis_light_high,) <span class="sc">&amp;</span> </span>
<span id="cb54-20"><a href="#cb54-20" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'right'</span>,</span>
<span id="cb54-21"><a href="#cb54-21" aria-hidden="true" tabindex="-1"></a>                              <span class="at">legend.box =</span> <span class="st">'horizontal'</span>,</span>
<span id="cb54-22"><a href="#cb54-22" aria-hidden="true" tabindex="-1"></a>                              <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">4</span>)))})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-1-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-1" role="tab" aria-controls="tabset-1-1" aria-selected="true" href="">genes_general</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-2" role="tab" aria-controls="tabset-1-2" aria-selected="false" href="">genes_arg1</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-3" role="tab" aria-controls="tabset-1-3" aria-selected="false" href="">genes_arg2</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-4-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-4" role="tab" aria-controls="tabset-1-4" aria-selected="false" href="">genes_nos2</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-5-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-5" role="tab" aria-controls="tabset-1-5" aria-selected="false" href="">genes_ifit1</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-6-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-6" role="tab" aria-controls="tabset-1-6" aria-selected="false" href="">genes_saa3</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-7-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-7" role="tab" aria-controls="tabset-1-7" aria-selected="false" href="">genes_cxcl10</a></li></ul>
<div class="tab-content">
<div id="tabset-1-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-1-1-tab">
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/plotMacStateMarkers-1.png" class="img-fluid" width="1440"></p>
</div>
</div>
<div id="tabset-1-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-2-tab">
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/plotMacStateMarkers-2.png" class="img-fluid" width="1440"></p>
</div>
</div>
<div id="tabset-1-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-3-tab">
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/plotMacStateMarkers-3.png" class="img-fluid" width="1440"></p>
</div>
</div>
<div id="tabset-1-4" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-4-tab">
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/plotMacStateMarkers-4.png" class="img-fluid" width="1440"></p>
</div>
</div>
<div id="tabset-1-5" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-5-tab">
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/plotMacStateMarkers-5.png" class="img-fluid" width="1440"></p>
</div>
</div>
<div id="tabset-1-6" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-6-tab">
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/plotMacStateMarkers-6.png" class="img-fluid" width="1440"></p>
</div>
</div>
<div id="tabset-1-7" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-7-tab">
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/plotMacStateMarkers-7.png" class="img-fluid" width="1440"></p>
</div>
</div>
</div>
</div>
<p>While there is no perfect match of the markers, we can still see a good correspondance our subclustering and the respective substate markers described by Giladi et al.&nbsp;It seems not to be possible to retrieve the the two individual Arg1 populations. according to these markers, our clusters correspond to: - 0: Arg1+ - 1: Arg1+ - 2: Nos2+ - 3: Cxcl10+ - 4: Saa3+ (by the markers other than Saa3) - 5: Ifi2+</p>
</section>
<section id="rename-subtypes" class="level3">
<h3 class="anchored" data-anchor-id="rename-subtypes">rename subtypes</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>mac.combined<span class="sc">$</span>clusters <span class="ot">=</span> mac.combined<span class="sc">$</span>seurat_clusters</span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(mac.combined) <span class="ot">=</span> mac.combined<span class="sc">$</span>clusters</span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a>mac.combined <span class="ot">=</span> <span class="fu">RenameIdents</span>(mac.combined,</span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a><span class="st">'0'</span> <span class="ot">=</span> <span class="st">'Arg1+'</span>,</span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a><span class="st">'1'</span> <span class="ot">=</span> <span class="st">'Arg1+'</span>,</span>
<span id="cb55-6"><a href="#cb55-6" aria-hidden="true" tabindex="-1"></a><span class="st">'2'</span> <span class="ot">=</span> <span class="st">'Nos2+'</span>,</span>
<span id="cb55-7"><a href="#cb55-7" aria-hidden="true" tabindex="-1"></a><span class="st">'3'</span> <span class="ot">=</span> <span class="st">'Cxcl10+'</span>,</span>
<span id="cb55-8"><a href="#cb55-8" aria-hidden="true" tabindex="-1"></a><span class="st">'4'</span> <span class="ot">=</span> <span class="st">'Saa3+'</span>, <span class="co"># (by other markers,)</span></span>
<span id="cb55-9"><a href="#cb55-9" aria-hidden="true" tabindex="-1"></a><span class="st">'5'</span> <span class="ot">=</span> <span class="st">'Ifi2+'</span>)</span>
<span id="cb55-10"><a href="#cb55-10" aria-hidden="true" tabindex="-1"></a>mac.combined<span class="sc">$</span>macTypes <span class="ot">=</span> <span class="fu">Idents</span>(mac.combined)</span>
<span id="cb55-11"><a href="#cb55-11" aria-hidden="true" tabindex="-1"></a><span class="fu">SaveH5Seurat</span>(mac.combined,<span class="at">filename =</span> <span class="fu">paste0</span>(outdir,<span class="st">'mac.combined.h5seurat'</span>),<span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="microglia" class="level2">
<h2 class="anchored" data-anchor-id="microglia">Microglia</h2>
<section id="integration-of-microglia" class="level3">
<h3 class="anchored" data-anchor-id="integration-of-microglia">Integration of Microglia</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>sc.split <span class="ot">=</span> <span class="fu">SplitObject</span>(sc.mg,<span class="at">split.by =</span> <span class="st">'sample'</span>)</span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a><span class="co"># normalize and identify variable features for each dataset independently</span></span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>sc.split <span class="ot">&lt;-</span> <span class="fu">lapply</span>(sc.split,<span class="cf">function</span>(x) {</span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(x)</span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(x, <span class="at">selection.method =</span> <span class="st">"vst"</span>, <span class="at">nfeatures =</span> <span class="dv">2000</span>)</span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb56-8"><a href="#cb56-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-9"><a href="#cb56-9" aria-hidden="true" tabindex="-1"></a><span class="co"># select features that are repeatedly variable across datasets for integration</span></span>
<span id="cb56-10"><a href="#cb56-10" aria-hidden="true" tabindex="-1"></a>features <span class="ot">&lt;-</span> <span class="fu">SelectIntegrationFeatures</span>(sc.split)</span>
<span id="cb56-11"><a href="#cb56-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-12"><a href="#cb56-12" aria-hidden="true" tabindex="-1"></a>mg.anchors <span class="ot">&lt;-</span> <span class="fu">FindIntegrationAnchors</span>(sc.split, <span class="at">anchor.features =</span> features)</span>
<span id="cb56-13"><a href="#cb56-13" aria-hidden="true" tabindex="-1"></a><span class="co"># this command creates an 'integrated' data assay</span></span>
<span id="cb56-14"><a href="#cb56-14" aria-hidden="true" tabindex="-1"></a>mg.combined <span class="ot">&lt;-</span> <span class="fu">IntegrateData</span>(<span class="at">anchorset =</span> mg.anchors)</span>
<span id="cb56-15"><a href="#cb56-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-16"><a href="#cb56-16" aria-hidden="true" tabindex="-1"></a><span class="fu">DefaultAssay</span>(mg.combined) <span class="ot">=</span> <span class="st">'integrated'</span></span>
<span id="cb56-17"><a href="#cb56-17" aria-hidden="true" tabindex="-1"></a>mg.combined <span class="ot">=</span> mg.combined <span class="sc">%&gt;%</span>         <span class="fu">ScaleData</span>() <span class="sc">%&gt;%</span></span>
<span id="cb56-18"><a href="#cb56-18" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">RunPCA</span>(<span class="at">npcs =</span> <span class="dv">30</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb56-19"><a href="#cb56-19" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">RunUMAP</span>(<span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb56-20"><a href="#cb56-20" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">FindNeighbors</span>(<span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb56-21"><a href="#cb56-21" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">FindClusters</span>(<span class="at">resolution =</span> <span class="fl">0.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 1446
Number of edges: 55041

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.7837
Number of communities: 7
Elapsed time: 0 seconds</code></pre>
</div>
</div>
</section>
<section id="check-for-contamination-in-microglia-sample" class="level3">
<h3 class="anchored" data-anchor-id="check-for-contamination-in-microglia-sample">check for contamination in Microglia sample</h3>
<div class="cell" data-fig.asp="1">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>mg.combined <span class="sc">%&gt;%</span> <span class="fu">DimPlot_scCustom</span>(<span class="at">label =</span> <span class="cn">TRUE</span>) <span class="sc">/</span> <span class="fu">wrap_plots</span>(</span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lapply</span>(genes_celltypes,<span class="cf">function</span>(x){</span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">FeaturePlot_scCustom</span>(<span class="at">seurat_object =</span> mg.combined,</span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">features =</span> x,</span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>                         <span class="at">colors_use =</span> viridis_light_high) <span class="sc">+</span> </span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a>      <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'right'</span>,</span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a>            <span class="at">legend.box =</span> <span class="st">'horizontal'</span>,</span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a>            <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">4</span>))</span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a>  })) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/unnamed-chunk-31-1.png" class="img-fluid" width="1440"></p>
</div>
</div>
<div class="cell" data-fig.asp="1">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a>markers_mg <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'Bhlhe41'</span>,<span class="st">'Gpr34'</span>, <span class="st">'Hexb'</span>, <span class="st">'Olfml3'</span>, <span class="st">'P2ry12'</span>,<span class="st">'P2ry13'</span>,<span class="st">'Sall1'</span>,<span class="st">'Serpine2'</span>,<span class="st">'Siglech'</span>,<span class="st">'Sparc'</span>)</span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>mg.combined <span class="sc">%&gt;%</span> <span class="fu">DimPlot_scCustom</span>(<span class="at">label =</span> <span class="cn">TRUE</span>) <span class="sc">/</span> <span class="fu">wrap_plots</span>(</span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lapply</span>(markers_mg,<span class="cf">function</span>(x){</span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">FeaturePlot_scCustom</span>(<span class="at">seurat_object =</span> mg.combined,</span>
<span id="cb59-5"><a href="#cb59-5" aria-hidden="true" tabindex="-1"></a>                         <span class="at">features =</span> x,</span>
<span id="cb59-6"><a href="#cb59-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">colors_use =</span> viridis_light_high) <span class="sc">+</span> </span>
<span id="cb59-7"><a href="#cb59-7" aria-hidden="true" tabindex="-1"></a>      <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'right'</span>,</span>
<span id="cb59-8"><a href="#cb59-8" aria-hidden="true" tabindex="-1"></a>            <span class="at">legend.box =</span> <span class="st">'horizontal'</span>,</span>
<span id="cb59-9"><a href="#cb59-9" aria-hidden="true" tabindex="-1"></a>            <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">4</span>))</span>
<span id="cb59-10"><a href="#cb59-10" aria-hidden="true" tabindex="-1"></a>  })) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/unnamed-chunk-32-1.png" class="img-fluid" width="1440"></p>
</div>
</div>
</section>
<section id="clean-contamiting-cells" class="level3">
<h3 class="anchored" data-anchor-id="clean-contamiting-cells">clean contamiting cells</h3>
<p>Cluster 6 seems to be a contamination as there are several markers from other celltypes expressed Cluster 5 seems to be a contamination as well as expression of sevarl Microglia markers is missing</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>mg.combined <span class="ot">=</span> mg.combined <span class="sc">%&gt;%</span>     <span class="fu">subset</span>(seurat_clusters <span class="sc">%!in%</span> <span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">6</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">ScaleData</span>() <span class="sc">%&gt;%</span></span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">RunPCA</span>(<span class="at">npcs =</span> <span class="dv">30</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">RunUMAP</span>(<span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">FindNeighbors</span>(<span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">FindClusters</span>(<span class="at">resolution =</span> <span class="fl">0.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 1382
Number of edges: 54150

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.7769
Number of communities: 5
Elapsed time: 0 seconds</code></pre>
</div>
</div>
</section>
<section id="identification-of-specific-microglia-cell-states" class="level3">
<h3 class="anchored" data-anchor-id="identification-of-specific-microglia-cell-states">Identification of specific Microglia cell states</h3>
<p>The following analysis of microglia subclusters is based on the findings by Jordao et al.&nbsp;2019. - In their analysis all MG populations expressed Bhlhe41, Gpr34, Hexb, Olfml3, P2ry12, P2ry13, Sall1, Serpine2, Siglech, and Sparc. - But daMG clusters showed lower expression of P2ry12, Maf,Slc2a5 and higher expression of Ccl2, Cxcl10, Ly86,and Mki67,indicating a proliferative capacity of daMG alongside the production of chemokines. - From the paper by Jordao et al.:The most inflammatory disease-associated microglial subsets (daMG2, daMG3, and daMG4) strongly downregulated the core microglial genes P2ry12, Tmem119,and Selplg and up-regulated Ly86. Both P2RY12 and TMEM119 immunoreactivities were clearly down-regulated within the core of spinal cord lesions, whereas CD162 (encoded by Selplg) was only weakly reduced. By contrast, microglial MD-1 (encoded by Ly86) was strongly up-regulated in the lesions. Because of their transcriptional profile and their P2RY12loTMEM119loMD-1hi phenotype, we determined that only daMG2, daMG3, and daMG4 localized within the lesion sites.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>genes_mg <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_mg =</span> <span class="fu">c</span>(<span class="st">'Bhlhe41'</span>,<span class="st">'Gpr34'</span>, <span class="st">'Hexb'</span>, <span class="st">'Olfml3'</span>, <span class="st">'P2ry12'</span>,<span class="st">'P2ry13'</span>,<span class="st">'Sall1'</span>,<span class="st">'Serpine2'</span>,<span class="st">'Siglech'</span>,<span class="st">'Sparc'</span>),</span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_daMG_Up =</span> <span class="fu">c</span>(<span class="st">'Ccl2'</span>, <span class="st">'Cxcl10'</span>,<span class="st">'Ly86'</span>,<span class="st">'Mki67'</span>),</span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_daMG_Down =</span> <span class="fu">c</span>(<span class="st">'P2ry12'</span>,<span class="st">'Tmem119'</span>, <span class="st">'Selplg'</span>,<span class="st">'Maf'</span>, <span class="st">'Slc2a5'</span>),</span>
<span id="cb62-5"><a href="#cb62-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_hMG =</span> <span class="fu">c</span>(<span class="st">'Gpr34'</span>,<span class="st">'Siglech'</span>,<span class="st">'P2ry12'</span>,<span class="st">'Tmem119'</span>),</span>
<span id="cb62-6"><a href="#cb62-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_repMG =</span> <span class="fu">c</span>(<span class="st">'Top2a'</span>,<span class="st">'Mki67'</span>),</span>
<span id="cb62-7"><a href="#cb62-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_daMG1 =</span> <span class="fu">c</span>(<span class="st">'Cd83'</span>,<span class="st">'Ctsd'</span>,<span class="st">'Cd81'</span>),</span>
<span id="cb62-8"><a href="#cb62-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_daMG2 =</span> <span class="fu">c</span>(<span class="st">'Ctsb'</span>,<span class="st">'Apoe'</span>,<span class="st">'B2m'</span>,<span class="st">'Cst7'</span>,<span class="st">'Cd74'</span>,<span class="st">'Ly86'</span>),</span>
<span id="cb62-9"><a href="#cb62-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_daMG3 =</span> <span class="fu">c</span>(<span class="st">'Cxcl10'</span>,<span class="st">'Tnf'</span>,<span class="st">'Ccl4'</span>,<span class="st">'Il1a'</span>,<span class="st">'B2m'</span>,<span class="st">'Ly86'</span>),</span>
<span id="cb62-10"><a href="#cb62-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_daMG4 =</span> <span class="fu">c</span>(<span class="st">'Itm2b'</span>,<span class="st">'Ctss'</span>,<span class="st">'Ccl5'</span>,<span class="st">'Naaa'</span>,<span class="st">'Ly86'</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a>mg_states_plots <span class="ot">=</span> <span class="fu">lapply</span>(genes_mg,<span class="cf">function</span>(x){</span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">wrap_plots</span>(<span class="at">design =</span> <span class="st">'ABBB</span></span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a><span class="st">                                           #BBB'</span>,</span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">DimPlot_scCustom</span>(<span class="at">seurat_object =</span> mg.combined,</span>
<span id="cb63-5"><a href="#cb63-5" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">group.by =</span> <span class="st">'seurat_clusters'</span>,</span>
<span id="cb63-6"><a href="#cb63-6" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">label =</span> <span class="cn">TRUE</span>,</span>
<span id="cb63-7"><a href="#cb63-7" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">label.size =</span> <span class="dv">10</span>), </span>
<span id="cb63-8"><a href="#cb63-8" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">FeaturePlot_scCustom</span>(<span class="at">seurat_object =</span> mg.combined,</span>
<span id="cb63-9"><a href="#cb63-9" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">features =</span> x,</span>
<span id="cb63-10"><a href="#cb63-10" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">colors_use =</span> viridis_light_high,) <span class="sc">&amp;</span> </span>
<span id="cb63-11"><a href="#cb63-11" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'right'</span>,</span>
<span id="cb63-12"><a href="#cb63-12" aria-hidden="true" tabindex="-1"></a>                              <span class="at">legend.box =</span> <span class="st">'horizontal'</span>,</span>
<span id="cb63-13"><a href="#cb63-13" aria-hidden="true" tabindex="-1"></a>                              <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">4</span>)))})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-2-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-1" role="tab" aria-controls="tabset-2-1" aria-selected="true" href="">genes_mg</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-2-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-2" role="tab" aria-controls="tabset-2-2" aria-selected="false" href="">genes_daMG_Up</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-2-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-3" role="tab" aria-controls="tabset-2-3" aria-selected="false" href="">genes_daMG_Down</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-2-4-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-4" role="tab" aria-controls="tabset-2-4" aria-selected="false" href="">genes_hMG</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-2-5-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-5" role="tab" aria-controls="tabset-2-5" aria-selected="false" href="">genes_repMG</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-2-6-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-6" role="tab" aria-controls="tabset-2-6" aria-selected="false" href="">genes_daMG1</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-2-7-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-7" role="tab" aria-controls="tabset-2-7" aria-selected="false" href="">genes_daMG2</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-2-8-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-8" role="tab" aria-controls="tabset-2-8" aria-selected="false" href="">genes_daMG3</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-2-9-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-9" role="tab" aria-controls="tabset-2-9" aria-selected="false" href="">genes_daMG4</a></li></ul>
<div class="tab-content">
<div id="tabset-2-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-2-1-tab">
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/plotMgStateMarkers-1.png" class="img-fluid" width="1440"></p>
</div>
</div>
<div id="tabset-2-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-2-2-tab">
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/plotMgStateMarkers-2.png" class="img-fluid" width="1440"></p>
</div>
</div>
<div id="tabset-2-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-2-3-tab">
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/plotMgStateMarkers-3.png" class="img-fluid" width="1440"></p>
</div>
</div>
<div id="tabset-2-4" class="tab-pane" role="tabpanel" aria-labelledby="tabset-2-4-tab">
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/plotMgStateMarkers-4.png" class="img-fluid" width="1440"></p>
</div>
</div>
<div id="tabset-2-5" class="tab-pane" role="tabpanel" aria-labelledby="tabset-2-5-tab">
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/plotMgStateMarkers-5.png" class="img-fluid" width="1440"></p>
</div>
</div>
<div id="tabset-2-6" class="tab-pane" role="tabpanel" aria-labelledby="tabset-2-6-tab">
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/plotMgStateMarkers-6.png" class="img-fluid" width="1440"></p>
</div>
</div>
<div id="tabset-2-7" class="tab-pane" role="tabpanel" aria-labelledby="tabset-2-7-tab">
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/plotMgStateMarkers-7.png" class="img-fluid" width="1440"></p>
</div>
</div>
<div id="tabset-2-8" class="tab-pane" role="tabpanel" aria-labelledby="tabset-2-8-tab">
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/plotMgStateMarkers-8.png" class="img-fluid" width="1440"></p>
</div>
</div>
<div id="tabset-2-9" class="tab-pane" role="tabpanel" aria-labelledby="tabset-2-9-tab">
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/plotMgStateMarkers-9.png" class="img-fluid" width="1440"></p>
</div>
</div>
</div>
</div>
<ul>
<li>All cells express most of the the micglia markers</li>
<li>Clusters 1,2,4 express more daMG associated genes</li>
<li>Cluster 0 containss hMG (has stronger expression in all genes downregulated in daMG and less expression in genes upregulated in daMG)</li>
<li>Cluster 3 contains Microglia with an active cell cylce</li>
<li>Genes that mark daMG1 don’t show variation across our data set so daMG1 population is probably not captured</li>
<li>Cluster 2 is most similar to daMG2</li>
<li>Cluster 1 seems to be an intermediate between hMG and daMG2, but also has some dMG3 genes expressed</li>
<li>Cluster 4 is most similar to daMG3</li>
<li>the daMG4 signature doesn’t show as a clear subpopulation, might be not well captured here</li>
</ul>
</section>
<section id="assign-mg-type-names" class="level3">
<h3 class="anchored" data-anchor-id="assign-mg-type-names">assign mg type names</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a>mg.combined<span class="sc">$</span>clusters <span class="ot">=</span> mg.combined<span class="sc">$</span>seurat_clusters</span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(mg.combined) <span class="ot">=</span> mg.combined<span class="sc">$</span>clusters</span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a>mg.combined <span class="ot">=</span> <span class="fu">RenameIdents</span>(mg.combined,</span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a><span class="st">'0'</span> <span class="ot">=</span> <span class="st">'hMG'</span>,</span>
<span id="cb64-5"><a href="#cb64-5" aria-hidden="true" tabindex="-1"></a><span class="st">'1'</span> <span class="ot">=</span> <span class="st">'hMG/daMG'</span>,</span>
<span id="cb64-6"><a href="#cb64-6" aria-hidden="true" tabindex="-1"></a><span class="st">'2'</span> <span class="ot">=</span> <span class="st">'daMG2'</span>,</span>
<span id="cb64-7"><a href="#cb64-7" aria-hidden="true" tabindex="-1"></a><span class="st">'3'</span> <span class="ot">=</span> <span class="st">'replicating MG'</span>,</span>
<span id="cb64-8"><a href="#cb64-8" aria-hidden="true" tabindex="-1"></a><span class="st">'4'</span> <span class="ot">=</span> <span class="st">'daMG3'</span>)</span>
<span id="cb64-9"><a href="#cb64-9" aria-hidden="true" tabindex="-1"></a>mg.combined<span class="sc">$</span>mgTypes <span class="ot">=</span> <span class="fu">Idents</span>(mg.combined)</span>
<span id="cb64-10"><a href="#cb64-10" aria-hidden="true" tabindex="-1"></a><span class="fu">SaveH5Seurat</span>(mg.combined,<span class="fu">paste0</span>(outdir,<span class="st">'mg.combined.h5seurat'</span>),<span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
</section>
<section id="figures" class="level1">
<h1>Figures</h1>
<p>The following blocks generate the figures as found in the paper.</p>
<section id="load-datasets" class="level2">
<h2 class="anchored" data-anchor-id="load-datasets">Load datasets</h2>
</section>
<section id="initialize-figures" class="level2">
<h2 class="anchored" data-anchor-id="initialize-figures">Initialize figures</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="co">#set graphical plot paramters </span></span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a>cols <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'#D4D4D4'</span>,<span class="st">'#CCF9E8'</span>)</span>
<span id="cb65-3"><a href="#cb65-3" aria-hidden="true" tabindex="-1"></a>colsDots <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'black'</span>,<span class="st">'#008000'</span>)</span>
<span id="cb65-4"><a href="#cb65-4" aria-hidden="true" tabindex="-1"></a>jWidth <span class="ot">=</span> <span class="fl">0.3</span></span>
<span id="cb65-5"><a href="#cb65-5" aria-hidden="true" tabindex="-1"></a>dWidth <span class="ot">=</span> <span class="fl">0.6</span></span>
<span id="cb65-6"><a href="#cb65-6" aria-hidden="true" tabindex="-1"></a>pSize <span class="ot">=</span> <span class="fl">0.2</span></span>
<span id="cb65-7"><a href="#cb65-7" aria-hidden="true" tabindex="-1"></a>pAlpha <span class="ot">=</span> <span class="fl">0.3</span> </span>
<span id="cb65-8"><a href="#cb65-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-9"><a href="#cb65-9" aria-hidden="true" tabindex="-1"></a>theme_umap <span class="ot">=</span> <span class="fu">theme</span>(<span class="at">axis.line =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb65-10"><a href="#cb65-10" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb65-11"><a href="#cb65-11" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.text =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb65-12"><a href="#cb65-12" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="dv">0</span>),</span>
<span id="cb65-13"><a href="#cb65-13" aria-hidden="true" tabindex="-1"></a>                              <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="dv">0</span>)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="overview" class="level2">
<h2 class="anchored" data-anchor-id="overview">Overview</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>sc.IC<span class="sc">$</span>sample <span class="ot">=</span> <span class="fu">factor</span>(sc.IC<span class="sc">$</span>sample, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">'EAE_IC_WT'</span>,<span class="st">'EAE_IC_HMBP'</span>))</span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>sc.IC<span class="sc">$</span>celltypesNew <span class="ot">=</span> sc.IC<span class="sc">$</span>celltypesNew <span class="sc">%&gt;%</span> <span class="fu">recode</span>(<span class="at">Monocytes =</span> <span class="st">'Activated DC'</span>)</span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a><span class="co">#changed SingleDimPlot using trace('SingleDimPlot',edit = TRUE) with geom_point(..., stroke = 0) to get finer detail </span></span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a>umapCelltypes <span class="ot">=</span> sc.IC <span class="sc">%&gt;%</span> <span class="fu">DimPlot_scCustom</span>(<span class="at">group.by =</span> <span class="st">'celltypesNew'</span>,</span>
<span id="cb66-6"><a href="#cb66-6" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">label =</span> <span class="cn">FALSE</span>,</span>
<span id="cb66-7"><a href="#cb66-7" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">pt.size =</span> <span class="fl">0.5</span>) <span class="sc">&amp;</span> </span>
<span id="cb66-8"><a href="#cb66-8" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">NoAxes</span>() <span class="sc">&amp;</span></span>
<span id="cb66-9"><a href="#cb66-9" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="fu">c</span>(<span class="fl">0.45</span>,<span class="fl">0.10</span>),</span>
<span id="cb66-10"><a href="#cb66-10" aria-hidden="true" tabindex="-1"></a>                                 <span class="co"># legend.position = 'none',</span></span>
<span id="cb66-11"><a href="#cb66-11" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">plot.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb66-12"><a href="#cb66-12" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">6</span>),</span>
<span id="cb66-13"><a href="#cb66-13" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">legend.key.size =</span> <span class="fu">unit</span>(<span class="fl">0.6</span>,<span class="st">'lines'</span>),</span>
<span id="cb66-14"><a href="#cb66-14" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))  <span class="sc">&amp;</span></span>
<span id="cb66-15"><a href="#cb66-15" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">guides</span>(<span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">2</span>,<span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">size =</span> <span class="dv">2</span>)))</span>
<span id="cb66-16"><a href="#cb66-16" aria-hidden="true" tabindex="-1"></a>umapSamples <span class="ot">=</span> sc.IC <span class="sc">%&gt;%</span> </span>
<span id="cb66-17"><a href="#cb66-17" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">DimPlot</span>(<span class="at">split.by =</span> <span class="st">'sample'</span>, </span>
<span id="cb66-18"><a href="#cb66-18" aria-hidden="true" tabindex="-1"></a>                  <span class="at">group.by =</span> <span class="st">'sample'</span>, </span>
<span id="cb66-19"><a href="#cb66-19" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cols =</span> colsDots,</span>
<span id="cb66-20"><a href="#cb66-20" aria-hidden="true" tabindex="-1"></a>                  <span class="at">pt.size =</span> <span class="fl">0.2</span>)  <span class="sc">&amp;</span> </span>
<span id="cb66-21"><a href="#cb66-21" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">NoAxes</span>() <span class="sc">&amp;</span> </span>
<span id="cb66-22"><a href="#cb66-22" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'none'</span>,</span>
<span id="cb66-23"><a href="#cb66-23" aria-hidden="true" tabindex="-1"></a>                        <span class="at">plot.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb66-24"><a href="#cb66-24" aria-hidden="true" tabindex="-1"></a>                        <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>))</span>
<span id="cb66-25"><a href="#cb66-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-26"><a href="#cb66-26" aria-hidden="true" tabindex="-1"></a>umap <span class="ot">=</span> umapCelltypes <span class="sc">+</span> <span class="fu">inset_element</span>(umapSamples,<span class="at">top =</span> <span class="dv">1</span>,</span>
<span id="cb66-27"><a href="#cb66-27" aria-hidden="true" tabindex="-1"></a>                                                 <span class="at">right =</span> <span class="dv">1</span>,</span>
<span id="cb66-28"><a href="#cb66-28" aria-hidden="true" tabindex="-1"></a>                                                 <span class="at">left =</span> <span class="fl">0.4</span>,</span>
<span id="cb66-29"><a href="#cb66-29" aria-hidden="true" tabindex="-1"></a>                                                 <span class="at">bottom =</span> <span class="fl">0.5</span>)</span>
<span id="cb66-30"><a href="#cb66-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-31"><a href="#cb66-31" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(figdir,<span class="st">'umap.svg'</span>),<span class="at">plot =</span> umap,<span class="at">dpi =</span> <span class="dv">600</span>,<span class="at">width =</span> <span class="fl">9.22</span>,<span class="at">height =</span> <span class="fl">8.2</span>,<span class="at">units =</span> <span class="st">'cm'</span>)</span>
<span id="cb66-32"><a href="#cb66-32" aria-hidden="true" tabindex="-1"></a>umap</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/unnamed-chunk-37-1.png" class="img-fluid" width="480"></p>
</div>
</div>
</section>
<section id="gene-expression-plots" class="level2">
<h2 class="anchored" data-anchor-id="gene-expression-plots">Gene expression plots</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a>VlnPlotTheme <span class="ot">=</span> <span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">axis.line.x =</span> <span class="fu">element_line</span>(),</span>
<span id="cb67-3"><a href="#cb67-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">8</span>),</span>
<span id="cb67-4"><a href="#cb67-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">8</span>,<span class="at">color =</span> <span class="st">'black'</span>))</span>
<span id="cb67-5"><a href="#cb67-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-6"><a href="#cb67-6" aria-hidden="true" tabindex="-1"></a><span class="co">#define datasets</span></span>
<span id="cb67-7"><a href="#cb67-7" aria-hidden="true" tabindex="-1"></a>sc.t  <span class="ot">=</span> sc.IC <span class="sc">%&gt;%</span> <span class="fu">subset</span>(celltypesNew <span class="sc">==</span> <span class="st">'T-Cells'</span>)</span>
<span id="cb67-8"><a href="#cb67-8" aria-hidden="true" tabindex="-1"></a>sc.mac <span class="ot">=</span> sc.IC <span class="sc">%&gt;%</span> <span class="fu">subset</span>(celltypesNew <span class="sc">==</span> <span class="st">'Macrophages'</span>)</span>
<span id="cb67-9"><a href="#cb67-9" aria-hidden="true" tabindex="-1"></a>sc.mg <span class="ot">=</span> sc.IC <span class="sc">%&gt;%</span> <span class="fu">subset</span>(celltypesNew <span class="sc">==</span> <span class="st">'Microglia'</span>)</span>
<span id="cb67-10"><a href="#cb67-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-11"><a href="#cb67-11" aria-hidden="true" tabindex="-1"></a><span class="co">#define markers</span></span>
<span id="cb67-12"><a href="#cb67-12" aria-hidden="true" tabindex="-1"></a><span class="co">#CD25 = Il2ra, CD69, CD71 = Tfrc,HLA-DR,Cd137 = Tnfrsf9, CD107a = Lamp1, </span></span>
<span id="cb67-13"><a href="#cb67-13" aria-hidden="true" tabindex="-1"></a>tActMarker <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'Il12a'</span>,<span class="st">'Ifng'</span>,<span class="st">'Spp1'</span>,<span class="st">'Il17a'</span>,<span class="st">'Cd44'</span>,<span class="st">'Cd69'</span>,<span class="st">'Ccl2'</span>)</span>
<span id="cb67-14"><a href="#cb67-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-15"><a href="#cb67-15" aria-hidden="true" tabindex="-1"></a>mgActMarker <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'Hexb'</span>,<span class="st">'Tmem119'</span>,<span class="st">'P2ry12'</span>,<span class="st">'Axl'</span>,<span class="st">'Cybb'</span>,<span class="st">'Cd68'</span>,<span class="st">'H2-Aa'</span>,<span class="st">'Cd74'</span>,<span class="st">'Cd86'</span>,<span class="st">'Trem2'</span>,<span class="st">'Tyrobp'</span>,<span class="st">'Apoe'</span>,<span class="st">'Itgax'</span>,<span class="st">'Lgals3'</span>)</span>
<span id="cb67-16"><a href="#cb67-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-17"><a href="#cb67-17" aria-hidden="true" tabindex="-1"></a>cytokines <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'Ccl5'</span>,<span class="st">'Ccl6'</span>,<span class="st">'Cxcl10'</span>,<span class="st">'Il6'</span>,<span class="st">'Csf2'</span>,<span class="st">'Tnf'</span>)</span>
<span id="cb67-18"><a href="#cb67-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-19"><a href="#cb67-19" aria-hidden="true" tabindex="-1"></a>macActGenes <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'Ly6c2'</span>,<span class="st">'Ccr2'</span>,<span class="st">'Fn1'</span>,<span class="st">'Cd44'</span>)</span>
<span id="cb67-20"><a href="#cb67-20" aria-hidden="true" tabindex="-1"></a>macHomGenes <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'Fcgr1'</span>,<span class="st">'Ms4a7'</span>,<span class="st">'Mertk'</span>)</span>
<span id="cb67-21"><a href="#cb67-21" aria-hidden="true" tabindex="-1"></a>macGenes <span class="ot">=</span> <span class="fu">c</span>(macHomGenes,macActGenes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="microglia-1" class="level3">
<h3 class="anchored" data-anchor-id="microglia-1">Microglia</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Microglia </span></span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a>dfMgWt <span class="ot">=</span> <span class="fu">FetchData</span>(sc.mg,<span class="at">vars =</span> mgActMarker, <span class="at">slot =</span> <span class="st">'data'</span>,<span class="at">cells =</span> <span class="fu">colnames</span>(sc.mg)[sc.mg<span class="sc">$</span>sample <span class="sc">==</span><span class="st">'EAE_IC_WT'</span>]) <span class="sc">%&gt;%</span> <span class="fu">gather</span>() <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">genotype =</span> <span class="st">'WT'</span>)</span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a>dfMgHmbp <span class="ot">=</span> <span class="fu">FetchData</span>(sc.mg,<span class="at">vars =</span> mgActMarker, <span class="at">slot =</span> <span class="st">'data'</span>,<span class="at">cells =</span> <span class="fu">colnames</span>(sc.mg)[sc.mg<span class="sc">$</span>sample <span class="sc">==</span><span class="st">'EAE_IC_HMBP'</span>]) <span class="sc">%&gt;%</span> <span class="fu">gather</span>() <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">genotype =</span> <span class="st">'hMBP'</span>)</span>
<span id="cb68-4"><a href="#cb68-4" aria-hidden="true" tabindex="-1"></a>dfMg <span class="ot">=</span> <span class="fu">rbind</span>(dfMgWt,dfMgHmbp)</span>
<span id="cb68-5"><a href="#cb68-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-6"><a href="#cb68-6" aria-hidden="true" tabindex="-1"></a>mgPlot <span class="ot">=</span> <span class="fu">ggplot</span>(dfMg, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(key,<span class="at">level =</span> mgActMarker), </span>
<span id="cb68-7"><a href="#cb68-7" aria-hidden="true" tabindex="-1"></a>                          <span class="at">y =</span> value,</span>
<span id="cb68-8"><a href="#cb68-8" aria-hidden="true" tabindex="-1"></a>                          <span class="at">fill =</span> <span class="fu">factor</span>(genotype,<span class="at">levels =</span> <span class="fu">c</span>(<span class="st">'WT'</span>,<span class="st">'hMBP'</span>)))) <span class="sc">+</span></span>
<span id="cb68-9"><a href="#cb68-9" aria-hidden="true" tabindex="-1"></a>                <span class="fu">geom_split_violin</span>(<span class="at">scale =</span> <span class="st">'width'</span>) <span class="sc">+</span> </span>
<span id="cb68-10"><a href="#cb68-10" aria-hidden="true" tabindex="-1"></a>                <span class="fu">geom_jitter</span>(<span class="at">position =</span> <span class="fu">position_jitterdodge</span>(</span>
<span id="cb68-11"><a href="#cb68-11" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">jitter.width =</span> jWidth, </span>
<span id="cb68-12"><a href="#cb68-12" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">dodge.width =</span> dWidth),</span>
<span id="cb68-13"><a href="#cb68-13" aria-hidden="true" tabindex="-1"></a>                            <span class="at">size =</span> pSize,</span>
<span id="cb68-14"><a href="#cb68-14" aria-hidden="true" tabindex="-1"></a>                            <span class="at">alpha =</span> pAlpha,</span>
<span id="cb68-15"><a href="#cb68-15" aria-hidden="true" tabindex="-1"></a>                            <span class="at">stroke =</span> <span class="dv">0</span>,</span>
<span id="cb68-16"><a href="#cb68-16" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">aes</span>(<span class="at">color =</span> <span class="fu">factor</span>(genotype,<span class="at">levels =</span> <span class="fu">c</span>(<span class="st">'WT'</span>,<span class="st">'hMBP'</span>)))) <span class="sc">+</span> </span>
<span id="cb68-17"><a href="#cb68-17" aria-hidden="true" tabindex="-1"></a>                <span class="fu">scale_fill_discrete</span>(<span class="at">type =</span> cols) <span class="sc">+</span> </span>
<span id="cb68-18"><a href="#cb68-18" aria-hidden="true" tabindex="-1"></a>                <span class="fu">scale_color_discrete</span>(<span class="at">type =</span> colsDots) <span class="sc">+</span> </span>
<span id="cb68-19"><a href="#cb68-19" aria-hidden="true" tabindex="-1"></a>                <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="dv">7</span>) <span class="sc">+</span></span>
<span id="cb68-20"><a href="#cb68-20" aria-hidden="true" tabindex="-1"></a>                <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb68-21"><a href="#cb68-21" aria-hidden="true" tabindex="-1"></a>                      <span class="at">legend.position =</span> <span class="st">'none'</span>,</span>
<span id="cb68-22"><a href="#cb68-22" aria-hidden="true" tabindex="-1"></a>                      <span class="at">axis.line.y =</span> <span class="fu">element_line</span>(),</span>
<span id="cb68-23"><a href="#cb68-23" aria-hidden="true" tabindex="-1"></a>                      <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span> </span>
<span id="cb68-24"><a href="#cb68-24" aria-hidden="true" tabindex="-1"></a>                <span class="fu">xlab</span>(<span class="st">'Microglia activation genes'</span>) <span class="sc">+</span> </span>
<span id="cb68-25"><a href="#cb68-25" aria-hidden="true" tabindex="-1"></a>                <span class="fu">ylab</span>(<span class="st">'Log1p-normalized expression'</span>) <span class="sc">+</span> </span>
<span id="cb68-26"><a href="#cb68-26" aria-hidden="true" tabindex="-1"></a>                VlnPlotTheme</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="t-cells" class="level3">
<h3 class="anchored" data-anchor-id="t-cells">T-cells</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="co">#T-Cells</span></span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a>dfTWt <span class="ot">=</span> <span class="fu">FetchData</span>(sc.t,<span class="at">vars =</span> tActMarker, <span class="at">slot =</span> <span class="st">'data'</span>,<span class="at">cells =</span> <span class="fu">colnames</span>(sc.t)[sc.t<span class="sc">$</span>sample <span class="sc">==</span><span class="st">'EAE_IC_WT'</span>]) <span class="sc">%&gt;%</span> <span class="fu">gather</span>() <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">genotype =</span> <span class="st">'WT'</span>)</span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a>dfTHmbp <span class="ot">=</span> <span class="fu">FetchData</span>(sc.t,<span class="at">vars =</span> tActMarker, <span class="at">slot =</span> <span class="st">'data'</span>,<span class="at">cells =</span> <span class="fu">colnames</span>(sc.t)[sc.t<span class="sc">$</span>sample <span class="sc">==</span><span class="st">'EAE_IC_HMBP'</span>]) <span class="sc">%&gt;%</span> <span class="fu">gather</span>() <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">genotype =</span> <span class="st">'hMBP'</span>)</span>
<span id="cb69-4"><a href="#cb69-4" aria-hidden="true" tabindex="-1"></a>dfT <span class="ot">=</span> <span class="fu">rbind</span>(dfTWt,dfTHmbp)</span>
<span id="cb69-5"><a href="#cb69-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-6"><a href="#cb69-6" aria-hidden="true" tabindex="-1"></a>tPlot <span class="ot">=</span> <span class="fu">ggplot</span>(dfT, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(key,<span class="at">level =</span> tActMarker), <span class="at">y =</span> value,<span class="at">fill =</span> <span class="fu">factor</span>(genotype,<span class="at">levels =</span> <span class="fu">c</span>(<span class="st">'WT'</span>,<span class="st">'hMBP'</span>)))) <span class="sc">+</span> </span>
<span id="cb69-7"><a href="#cb69-7" aria-hidden="true" tabindex="-1"></a>              <span class="fu">geom_split_violin</span>(<span class="at">scale =</span> <span class="st">'width'</span>) <span class="sc">+</span> </span>
<span id="cb69-8"><a href="#cb69-8" aria-hidden="true" tabindex="-1"></a>                <span class="fu">geom_jitter</span>(<span class="at">position =</span> <span class="fu">position_jitterdodge</span>(</span>
<span id="cb69-9"><a href="#cb69-9" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">jitter.width =</span> jWidth, </span>
<span id="cb69-10"><a href="#cb69-10" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">dodge.width =</span> dWidth),</span>
<span id="cb69-11"><a href="#cb69-11" aria-hidden="true" tabindex="-1"></a>                            <span class="at">size =</span> pSize,</span>
<span id="cb69-12"><a href="#cb69-12" aria-hidden="true" tabindex="-1"></a>                            <span class="at">alpha =</span> pAlpha,</span>
<span id="cb69-13"><a href="#cb69-13" aria-hidden="true" tabindex="-1"></a>                            <span class="at">stroke =</span> <span class="dv">0</span>,</span>
<span id="cb69-14"><a href="#cb69-14" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">aes</span>(<span class="at">color =</span> <span class="fu">factor</span>(genotype,<span class="at">levels =</span> <span class="fu">c</span>(<span class="st">'WT'</span>,<span class="st">'hMBP'</span>)))) <span class="sc">+</span> </span>
<span id="cb69-15"><a href="#cb69-15" aria-hidden="true" tabindex="-1"></a>              <span class="fu">scale_fill_discrete</span>(<span class="st">'Genotype'</span>,<span class="at">type =</span> cols) <span class="sc">+</span> </span>
<span id="cb69-16"><a href="#cb69-16" aria-hidden="true" tabindex="-1"></a>              <span class="fu">scale_color_discrete</span>(<span class="st">'Genotype'</span>,<span class="at">type =</span> colsDots) <span class="sc">+</span> </span>
<span id="cb69-17"><a href="#cb69-17" aria-hidden="true" tabindex="-1"></a>              <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="dv">7</span>) <span class="sc">+</span> </span>
<span id="cb69-18"><a href="#cb69-18" aria-hidden="true" tabindex="-1"></a>              <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb69-19"><a href="#cb69-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb69-20"><a href="#cb69-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb69-21"><a href="#cb69-21" aria-hidden="true" tabindex="-1"></a>                    <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb69-22"><a href="#cb69-22" aria-hidden="true" tabindex="-1"></a>                    <span class="at">axis.line.y.left =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb69-23"><a href="#cb69-23" aria-hidden="true" tabindex="-1"></a>                    <span class="at">legend.position =</span> <span class="st">'none'</span>) <span class="sc">+</span> </span>
<span id="cb69-24"><a href="#cb69-24" aria-hidden="true" tabindex="-1"></a>              <span class="fu">xlab</span>(<span class="st">'T-Cell activation genes'</span>) <span class="sc">+</span> </span>
<span id="cb69-25"><a href="#cb69-25" aria-hidden="true" tabindex="-1"></a>              VlnPlotTheme</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="cytokines" class="level3">
<h3 class="anchored" data-anchor-id="cytokines">Cytokines</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a>dfCkWt <span class="ot">=</span> <span class="fu">FetchData</span>(sc.IC,<span class="at">vars =</span> cytokines, <span class="at">slot =</span> <span class="st">'data'</span>,<span class="at">cells =</span> <span class="fu">colnames</span>(sc.IC)[sc.IC<span class="sc">$</span>sample <span class="sc">==</span><span class="st">'EAE_IC_WT'</span>]) <span class="sc">%&gt;%</span> <span class="fu">gather</span>() <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">genotype =</span> <span class="st">'WT'</span>)</span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a>dfCkHmbp <span class="ot">=</span> <span class="fu">FetchData</span>(sc.IC,<span class="at">vars =</span> cytokines, <span class="at">slot =</span> <span class="st">'data'</span>,<span class="at">cells =</span> <span class="fu">colnames</span>(sc.IC)[sc.IC<span class="sc">$</span>sample <span class="sc">==</span><span class="st">'EAE_IC_HMBP'</span>]) <span class="sc">%&gt;%</span> <span class="fu">gather</span>() <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">genotype =</span> <span class="st">'hMBP'</span>)</span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a>dfCk <span class="ot">=</span> <span class="fu">rbind</span>(dfCkWt,dfCkHmbp)</span>
<span id="cb70-4"><a href="#cb70-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-5"><a href="#cb70-5" aria-hidden="true" tabindex="-1"></a>ckPlot <span class="ot">=</span> <span class="fu">ggplot</span>(dfCk, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(key,<span class="at">level =</span> cytokines), </span>
<span id="cb70-6"><a href="#cb70-6" aria-hidden="true" tabindex="-1"></a>                          <span class="at">y =</span> value,</span>
<span id="cb70-7"><a href="#cb70-7" aria-hidden="true" tabindex="-1"></a>                          <span class="at">fill =</span> <span class="fu">factor</span>(genotype,<span class="at">levels =</span> <span class="fu">c</span>(<span class="st">'WT'</span>,<span class="st">'hMBP'</span>)))) <span class="sc">+</span></span>
<span id="cb70-8"><a href="#cb70-8" aria-hidden="true" tabindex="-1"></a>                <span class="fu">geom_split_violin</span>(<span class="at">scale =</span> <span class="st">'width'</span>) <span class="sc">+</span> </span>
<span id="cb70-9"><a href="#cb70-9" aria-hidden="true" tabindex="-1"></a>                <span class="fu">geom_jitter</span>(<span class="at">position =</span> <span class="fu">position_jitterdodge</span>(</span>
<span id="cb70-10"><a href="#cb70-10" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">jitter.width =</span> jWidth, </span>
<span id="cb70-11"><a href="#cb70-11" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">dodge.width =</span> dWidth),</span>
<span id="cb70-12"><a href="#cb70-12" aria-hidden="true" tabindex="-1"></a>                            <span class="at">size =</span> pSize,</span>
<span id="cb70-13"><a href="#cb70-13" aria-hidden="true" tabindex="-1"></a>                            <span class="at">alpha =</span> pAlpha,</span>
<span id="cb70-14"><a href="#cb70-14" aria-hidden="true" tabindex="-1"></a>                            <span class="at">stroke =</span> <span class="dv">0</span>,</span>
<span id="cb70-15"><a href="#cb70-15" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">aes</span>(<span class="at">color =</span> <span class="fu">factor</span>(genotype,<span class="at">levels =</span> <span class="fu">c</span>(<span class="st">'WT'</span>,<span class="st">'hMBP'</span>)))) <span class="sc">+</span> </span>
<span id="cb70-16"><a href="#cb70-16" aria-hidden="true" tabindex="-1"></a>                <span class="fu">scale_fill_discrete</span>(<span class="st">'Genotype'</span>,<span class="at">type =</span> cols) <span class="sc">+</span> </span>
<span id="cb70-17"><a href="#cb70-17" aria-hidden="true" tabindex="-1"></a>                <span class="fu">scale_color_discrete</span>(<span class="st">'Genotype'</span>,<span class="at">type =</span> colsDots) <span class="sc">+</span> </span>
<span id="cb70-18"><a href="#cb70-18" aria-hidden="true" tabindex="-1"></a>                <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="dv">7</span>) <span class="sc">+</span></span>
<span id="cb70-19"><a href="#cb70-19" aria-hidden="true" tabindex="-1"></a>                <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb70-20"><a href="#cb70-20" aria-hidden="true" tabindex="-1"></a>                      <span class="at">legend.position =</span> <span class="st">'right'</span>,</span>
<span id="cb70-21"><a href="#cb70-21" aria-hidden="true" tabindex="-1"></a>                      <span class="at">axis.line.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb70-22"><a href="#cb70-22" aria-hidden="true" tabindex="-1"></a>                      <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb70-23"><a href="#cb70-23" aria-hidden="true" tabindex="-1"></a>                      <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb70-24"><a href="#cb70-24" aria-hidden="true" tabindex="-1"></a>                      <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span> </span>
<span id="cb70-25"><a href="#cb70-25" aria-hidden="true" tabindex="-1"></a>                <span class="fu">xlab</span>(<span class="st">'Cytokines'</span>) <span class="sc">+</span> </span>
<span id="cb70-26"><a href="#cb70-26" aria-hidden="true" tabindex="-1"></a>                <span class="fu">ylab</span>(<span class="st">'Log1p-normalized expression'</span>) <span class="sc">+</span>  <span class="co">#as ln(counts/counts per cell * 10e4 +1</span></span>
<span id="cb70-27"><a href="#cb70-27" aria-hidden="true" tabindex="-1"></a>                VlnPlotTheme</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="macrophages-1" class="level3">
<h3 class="anchored" data-anchor-id="macrophages-1">Macrophages</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a>dfMacWt <span class="ot">=</span> <span class="fu">FetchData</span>(sc.mac,<span class="at">vars =</span> macGenes, <span class="at">slot =</span> <span class="st">'data'</span>,<span class="at">cells =</span> <span class="fu">colnames</span>(sc.mac)[sc.mac<span class="sc">$</span>sample <span class="sc">==</span><span class="st">'EAE_IC_WT'</span>]) <span class="sc">%&gt;%</span> <span class="fu">gather</span>() <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">genotype =</span> <span class="st">'WT'</span>)</span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a>dfMacHmbp <span class="ot">=</span> <span class="fu">FetchData</span>(sc.mac,<span class="at">vars =</span> macGenes, <span class="at">slot =</span> <span class="st">'data'</span>,<span class="at">cells =</span> <span class="fu">colnames</span>(sc.mac)[sc.mac<span class="sc">$</span>sample <span class="sc">==</span><span class="st">'EAE_IC_HMBP'</span>]) <span class="sc">%&gt;%</span> <span class="fu">gather</span>() <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">genotype =</span> <span class="st">'hMBP'</span>)</span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a>dfMac <span class="ot">=</span> <span class="fu">rbind</span>(dfMacWt,dfMacHmbp)</span>
<span id="cb71-4"><a href="#cb71-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-5"><a href="#cb71-5" aria-hidden="true" tabindex="-1"></a>macPlot <span class="ot">=</span> <span class="fu">ggplot</span>(dfMac, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(key,<span class="at">level =</span> macGenes), </span>
<span id="cb71-6"><a href="#cb71-6" aria-hidden="true" tabindex="-1"></a>                          <span class="at">y =</span> value,</span>
<span id="cb71-7"><a href="#cb71-7" aria-hidden="true" tabindex="-1"></a>                          <span class="at">fill =</span> <span class="fu">factor</span>(genotype,<span class="at">levels =</span> <span class="fu">c</span>(<span class="st">'WT'</span>,<span class="st">'hMBP'</span>)))) <span class="sc">+</span></span>
<span id="cb71-8"><a href="#cb71-8" aria-hidden="true" tabindex="-1"></a>                <span class="fu">geom_split_violin</span>(<span class="at">scale =</span> <span class="st">'width'</span>) <span class="sc">+</span> </span>
<span id="cb71-9"><a href="#cb71-9" aria-hidden="true" tabindex="-1"></a>                <span class="fu">geom_jitter</span>(<span class="at">position =</span> <span class="fu">position_jitterdodge</span>(</span>
<span id="cb71-10"><a href="#cb71-10" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">jitter.width =</span> jWidth, </span>
<span id="cb71-11"><a href="#cb71-11" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">dodge.width =</span> dWidth),</span>
<span id="cb71-12"><a href="#cb71-12" aria-hidden="true" tabindex="-1"></a>                            <span class="at">size =</span> pSize,</span>
<span id="cb71-13"><a href="#cb71-13" aria-hidden="true" tabindex="-1"></a>                            <span class="at">alpha =</span> pAlpha,</span>
<span id="cb71-14"><a href="#cb71-14" aria-hidden="true" tabindex="-1"></a>                            <span class="at">stroke =</span> <span class="dv">0</span>,</span>
<span id="cb71-15"><a href="#cb71-15" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">aes</span>(<span class="at">color =</span> <span class="fu">factor</span>(genotype,<span class="at">levels =</span> <span class="fu">c</span>(<span class="st">'WT'</span>,<span class="st">'hMBP'</span>)))) <span class="sc">+</span> </span>
<span id="cb71-16"><a href="#cb71-16" aria-hidden="true" tabindex="-1"></a>                <span class="fu">scale_fill_discrete</span>(<span class="st">'Genotype'</span>,<span class="at">type =</span> cols) <span class="sc">+</span> </span>
<span id="cb71-17"><a href="#cb71-17" aria-hidden="true" tabindex="-1"></a>                <span class="fu">scale_color_discrete</span>(<span class="st">'Genotype'</span>,<span class="at">type =</span> colsDots) <span class="sc">+</span> </span>
<span id="cb71-18"><a href="#cb71-18" aria-hidden="true" tabindex="-1"></a>                <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="dv">7</span>) <span class="sc">+</span></span>
<span id="cb71-19"><a href="#cb71-19" aria-hidden="true" tabindex="-1"></a>                <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb71-20"><a href="#cb71-20" aria-hidden="true" tabindex="-1"></a>                      <span class="at">legend.position =</span> <span class="st">'none'</span>,</span>
<span id="cb71-21"><a href="#cb71-21" aria-hidden="true" tabindex="-1"></a>                      <span class="at">axis.line.y =</span> <span class="fu">element_line</span>(),</span>
<span id="cb71-22"><a href="#cb71-22" aria-hidden="true" tabindex="-1"></a>                      <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.85</span> )) <span class="sc">+</span> </span>
<span id="cb71-23"><a href="#cb71-23" aria-hidden="true" tabindex="-1"></a>                <span class="fu">xlab</span>(<span class="st">'Macrophage activation genes'</span>) <span class="sc">+</span> </span>
<span id="cb71-24"><a href="#cb71-24" aria-hidden="true" tabindex="-1"></a>                <span class="fu">ylab</span>(<span class="st">'Log1p-normalized expression'</span>) <span class="sc">+</span>  <span class="co">#as ln(counts/counts per cell * 10e4 +1</span></span>
<span id="cb71-25"><a href="#cb71-25" aria-hidden="true" tabindex="-1"></a>                VlnPlotTheme</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="output-fig.-x" class="level3">
<h3 class="anchored" data-anchor-id="output-fig.-x">Output Fig. X</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a>mgPlot <span class="sc">%&gt;%</span> <span class="fu">ggsave</span>(<span class="fu">paste0</span>(figdir,<span class="st">'mg.svg'</span>), <span class="at">plot =</span> ., <span class="at">dpi =</span> <span class="dv">600</span>, <span class="at">width =</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="dv">4</span>, <span class="at">units =</span> <span class="st">'cm'</span>)</span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a>mgPlot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/unnamed-chunk-43-1.png" class="img-fluid" width="480"></p>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a>(macPlot <span class="sc">+</span> tPlot <span class="sc">+</span> ckPlot) <span class="sc">%&gt;%</span> <span class="fu">ggsave</span>(<span class="fu">paste0</span>(figdir,<span class="st">'t_mac_ck.svg'</span>),<span class="at">plot =</span> .,<span class="at">dpi =</span> <span class="dv">600</span>,<span class="at">width =</span> <span class="dv">18</span>,<span class="at">height =</span> <span class="dv">4</span>,<span class="at">units =</span> <span class="st">'cm'</span>)</span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-3"><a href="#cb73-3" aria-hidden="true" tabindex="-1"></a>(macPlot <span class="sc">+</span> tPlot <span class="sc">+</span> ckPlot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/unnamed-chunk-44-1.png" class="img-fluid" width="864"></p>
</div>
</div>
</section>
</section>
</section>
<section id="extended-figures" class="level1">
<h1>Extended Figures</h1>
<section id="qc" class="level2">
<h2 class="anchored" data-anchor-id="qc">QC</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a>themeVln <span class="ot">=</span> <span class="fu">theme_classic</span>() <span class="sc">+</span> </span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a>           <span class="fu">theme</span>(</span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">legend.position =</span> <span class="st">'none'</span>,</span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>)</span>
<span id="cb74-5"><a href="#cb74-5" aria-hidden="true" tabindex="-1"></a>                )</span>
<span id="cb74-6"><a href="#cb74-6" aria-hidden="true" tabindex="-1"></a>pSize <span class="ot">=</span> <span class="fl">0.3</span></span>
<span id="cb74-7"><a href="#cb74-7" aria-hidden="true" tabindex="-1"></a>lwd <span class="ot">=</span> <span class="fl">0.2</span></span>
<span id="cb74-8"><a href="#cb74-8" aria-hidden="true" tabindex="-1"></a>bpWidth <span class="ot">=</span> <span class="fl">0.1</span></span>
<span id="cb74-9"><a href="#cb74-9" aria-hidden="true" tabindex="-1"></a>linetype <span class="ot">=</span> <span class="st">'dashed'</span></span>
<span id="cb74-10"><a href="#cb74-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-11"><a href="#cb74-11" aria-hidden="true" tabindex="-1"></a>qc.counts.violin <span class="ot">=</span> sc.tagged<span class="sc">@</span>meta.data <span class="sc">%&gt;%</span> </span>
<span id="cb74-12"><a href="#cb74-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>genotype, <span class="at">y=</span>nCount_RNA, <span class="at">fill=</span> genotype, <span class="at">color =</span> genotype)) <span class="sc">+</span> </span>
<span id="cb74-13"><a href="#cb74-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_violin</span>(<span class="at">scale =</span> <span class="st">'width'</span>,<span class="at">lwd =</span> lwd) <span class="sc">+</span> </span>
<span id="cb74-14"><a href="#cb74-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>(<span class="at">size =</span> pSize, <span class="at">alpha =</span> pAlpha, <span class="at">stroke =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb74-15"><a href="#cb74-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_discrete</span>(<span class="st">'Genotpye'</span>, <span class="at">type =</span> colsDots) <span class="sc">+</span> </span>
<span id="cb74-16"><a href="#cb74-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_discrete</span>(<span class="st">'Genotpye'</span>,<span class="at">type =</span> cols)  <span class="sc">+</span> </span>
<span id="cb74-17"><a href="#cb74-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">width =</span> bpWidth, <span class="at">outlier.size =</span> <span class="dv">0</span>, <span class="at">coef =</span> <span class="dv">0</span>, <span class="at">outlier.color =</span> <span class="cn">NA</span>,<span class="at">lwd =</span> lwd) <span class="sc">+</span></span>
<span id="cb74-18"><a href="#cb74-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_log10</span>() <span class="sc">+</span> </span>
<span id="cb74-19"><a href="#cb74-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> cutoff.reads, <span class="at">linetype=</span>linetype) <span class="sc">+</span> </span>
<span id="cb74-20"><a href="#cb74-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">'Read Counts'</span>) <span class="sc">+</span></span>
<span id="cb74-21"><a href="#cb74-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">''</span>) <span class="sc">+</span> </span>
<span id="cb74-22"><a href="#cb74-22" aria-hidden="true" tabindex="-1"></a>    themeVln</span>
<span id="cb74-23"><a href="#cb74-23" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb74-24"><a href="#cb74-24" aria-hidden="true" tabindex="-1"></a>qc.features.violin <span class="ot">=</span> sc.tagged<span class="sc">@</span>meta.data <span class="sc">%&gt;%</span> </span>
<span id="cb74-25"><a href="#cb74-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>genotype, <span class="at">y=</span>nFeature_RNA, <span class="at">fill=</span> genotype, <span class="at">color =</span> genotype)) <span class="sc">+</span> </span>
<span id="cb74-26"><a href="#cb74-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_violin</span>(<span class="at">scale =</span> <span class="st">'width'</span>,<span class="at">lwd =</span> lwd) <span class="sc">+</span> </span>
<span id="cb74-27"><a href="#cb74-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>(<span class="at">size =</span> pSize, <span class="at">alpha =</span> pAlpha, <span class="at">stroke =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb74-28"><a href="#cb74-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_discrete</span>(<span class="st">'Genotype'</span>, <span class="at">type =</span> colsDots) <span class="sc">+</span> </span>
<span id="cb74-29"><a href="#cb74-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_discrete</span>(<span class="st">'Genotype'</span>,<span class="at">type =</span> cols)  <span class="sc">+</span> </span>
<span id="cb74-30"><a href="#cb74-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">width =</span> bpWidth, <span class="at">outlier.size =</span> <span class="dv">0</span>, <span class="at">coef =</span> <span class="dv">0</span>, <span class="at">outlier.color =</span> <span class="cn">NA</span>,<span class="at">lwd =</span> lwd) <span class="sc">+</span></span>
<span id="cb74-31"><a href="#cb74-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_log10</span>() <span class="sc">+</span> </span>
<span id="cb74-32"><a href="#cb74-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> cutoff.features, <span class="at">linetype =</span> linetype) <span class="sc">+</span> </span>
<span id="cb74-33"><a href="#cb74-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">'Feature Counts'</span>) <span class="sc">+</span></span>
<span id="cb74-34"><a href="#cb74-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">''</span>) <span class="sc">+</span> </span>
<span id="cb74-35"><a href="#cb74-35" aria-hidden="true" tabindex="-1"></a>    themeVln</span>
<span id="cb74-36"><a href="#cb74-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-37"><a href="#cb74-37" aria-hidden="true" tabindex="-1"></a>qc.mitoRatio.violin <span class="ot">=</span> sc.tagged<span class="sc">@</span>meta.data <span class="sc">%&gt;%</span> </span>
<span id="cb74-38"><a href="#cb74-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>genotype, <span class="at">y=</span>mitoRatio, <span class="at">fill=</span> genotype, <span class="at">color =</span> genotype)) <span class="sc">+</span> </span>
<span id="cb74-39"><a href="#cb74-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_violin</span>(<span class="at">scale =</span> <span class="st">'width'</span>,<span class="at">lwd =</span> lwd) <span class="sc">+</span> </span>
<span id="cb74-40"><a href="#cb74-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>(<span class="at">size =</span> pSize, <span class="at">alpha =</span> pAlpha, <span class="at">stroke =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb74-41"><a href="#cb74-41" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_discrete</span>(<span class="st">'Genotpye'</span>, <span class="at">type =</span> colsDots) <span class="sc">+</span> </span>
<span id="cb74-42"><a href="#cb74-42" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_discrete</span>(<span class="st">'Genotpye'</span>,<span class="at">type =</span> cols)  <span class="sc">+</span> </span>
<span id="cb74-43"><a href="#cb74-43" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">width =</span> bpWidth, <span class="at">outlier.size =</span> <span class="dv">0</span>, <span class="at">coef =</span> <span class="dv">0</span>, <span class="at">outlier.color =</span> <span class="cn">NA</span>,<span class="at">lwd =</span> lwd) <span class="sc">+</span></span>
<span id="cb74-44"><a href="#cb74-44" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_log10</span>() <span class="sc">+</span> </span>
<span id="cb74-45"><a href="#cb74-45" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> cutoff.mitoRatio, <span class="at">linetype=</span>linetype) <span class="sc">+</span> </span>
<span id="cb74-46"><a href="#cb74-46" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">'Mitochondrial Read Ratio'</span>) <span class="sc">+</span></span>
<span id="cb74-47"><a href="#cb74-47" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">''</span>) <span class="sc">+</span> </span>
<span id="cb74-48"><a href="#cb74-48" aria-hidden="true" tabindex="-1"></a>    themeVln</span>
<span id="cb74-49"><a href="#cb74-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-50"><a href="#cb74-50" aria-hidden="true" tabindex="-1"></a>    qcplot <span class="ot">=</span> qc.counts.violin <span class="sc">/</span> qc.features.violin <span class="sc">/</span> qc.mitoRatio.violin</span>
<span id="cb74-51"><a href="#cb74-51" aria-hidden="true" tabindex="-1"></a>qcplot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/unnamed-chunk-45-1.png" class="img-fluid" width="384"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(figdir,<span class="st">'qc_plots.svg'</span>),qcplot,<span class="at">dpi =</span> <span class="dv">600</span>,<span class="at">width =</span> <span class="dv">4</span>,<span class="at">height =</span> <span class="dv">12</span>,<span class="at">units =</span> <span class="st">'cm'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="pca-1" class="level2">
<h2 class="anchored" data-anchor-id="pca-1">PCA</h2>
<p>Plotting the biplots for first 10 PCs in pairs. No obvious separation is visible based on sample identity.</p>
<div class="cell" data-fig.asp="1">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="co">#with PCAtools to do some deeper analyses </span></span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a>VarFeatNorm <span class="ot">=</span> sc.IC<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>scale.data[<span class="fu">VariableFeatures</span>(sc.IC),]</span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-5"><a href="#cb76-5" aria-hidden="true" tabindex="-1"></a><span class="co">#shuffle colnames around to avoid overplotting in the biplot</span></span>
<span id="cb76-6"><a href="#cb76-6" aria-hidden="true" tabindex="-1"></a>x <span class="ot">=</span> <span class="fu">colnames</span>(VarFeatNorm)</span>
<span id="cb76-7"><a href="#cb76-7" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb76-8"><a href="#cb76-8" aria-hidden="true" tabindex="-1"></a>x <span class="ot">=</span> <span class="fu">sample</span>(x)</span>
<span id="cb76-9"><a href="#cb76-9" aria-hidden="true" tabindex="-1"></a>VarFeatNorm <span class="ot">=</span> VarFeatNorm[,x]</span>
<span id="cb76-10"><a href="#cb76-10" aria-hidden="true" tabindex="-1"></a>metaData <span class="ot">=</span> sc.IC<span class="sc">@</span>meta.data[x,]</span>
<span id="cb76-11"><a href="#cb76-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-12"><a href="#cb76-12" aria-hidden="true" tabindex="-1"></a><span class="co">#run PCA</span></span>
<span id="cb76-13"><a href="#cb76-13" aria-hidden="true" tabindex="-1"></a>p <span class="ot">=</span> PCAtools<span class="sc">::</span><span class="fu">pca</span>(VarFeatNorm,<span class="at">rank =</span> <span class="dv">10</span>,<span class="at">metadata =</span> metaData)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-fig.asp="1">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb77"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a>colr <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'grey'</span>,<span class="st">'darkgreen'</span>)</span>
<span id="cb77-2"><a href="#cb77-2" aria-hidden="true" tabindex="-1"></a>cols <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'#D4D4D4'</span>,<span class="st">'#CCF9E8'</span>)</span>
<span id="cb77-3"><a href="#cb77-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb77-4"><a href="#cb77-4" aria-hidden="true" tabindex="-1"></a>pcaPlot <span class="ot">=</span> <span class="fu">pairsplot</span>(p,</span>
<span id="cb77-5"><a href="#cb77-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">colby =</span> <span class="st">'sample'</span>,</span>
<span id="cb77-6"><a href="#cb77-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">components =</span> <span class="fu">getComponents</span>(p, <span class="fu">seq_len</span>(<span class="dv">10</span>)), </span>
<span id="cb77-7"><a href="#cb77-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">gridlines.major =</span> <span class="cn">FALSE</span>, </span>
<span id="cb77-8"><a href="#cb77-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">gridlines.minor =</span> <span class="cn">FALSE</span>, </span>
<span id="cb77-9"><a href="#cb77-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb77-10"><a href="#cb77-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">hline =</span> <span class="dv">0</span>, </span>
<span id="cb77-11"><a href="#cb77-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">vline =</span> <span class="dv">0</span>, </span>
<span id="cb77-12"><a href="#cb77-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">plotaxes =</span> <span class="cn">FALSE</span>,</span>
<span id="cb77-13"><a href="#cb77-13" aria-hidden="true" tabindex="-1"></a>                    <span class="at">pointSize =</span> <span class="fl">0.1</span>, </span>
<span id="cb77-14"><a href="#cb77-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">colkey =</span> colr) <span class="sc">&amp;</span> </span>
<span id="cb77-15"><a href="#cb77-15" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>))</span>
<span id="cb77-16"><a href="#cb77-16" aria-hidden="true" tabindex="-1"></a>pcaPlot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/PCA-1.png" class="img-fluid" width="864"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(figdir,<span class="st">'pca_pairsplot.svg'</span>),pcaPlot,<span class="at">dpi =</span> <span class="dv">600</span>,<span class="at">width =</span> <span class="dv">27</span>,<span class="at">height =</span> <span class="dv">27</span>,<span class="at">units =</span> <span class="st">'cm'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="celltypemarkers" class="level2">
<h2 class="anchored" data-anchor-id="celltypemarkers">CellTypeMarkers</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a>CellTypeMarkers <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a>                    <span class="st">'Cd19'</span>,<span class="st">'Ms4a1'</span>,</span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a>                    <span class="st">'S100a8'</span>,<span class="st">'Retnlg'</span>,</span>
<span id="cb79-4"><a href="#cb79-4" aria-hidden="true" tabindex="-1"></a>                    <span class="st">'Siglech'</span>,<span class="st">'Bst2'</span>,</span>
<span id="cb79-5"><a href="#cb79-5" aria-hidden="true" tabindex="-1"></a>                    <span class="st">'Ccr7'</span>,<span class="st">'Fscn1'</span>,</span>
<span id="cb79-6"><a href="#cb79-6" aria-hidden="true" tabindex="-1"></a>                    <span class="st">'Itgax'</span>,<span class="st">'Btla'</span>,</span>
<span id="cb79-7"><a href="#cb79-7" aria-hidden="true" tabindex="-1"></a>                    <span class="st">'Cd3d'</span>,<span class="st">'Trbc2'</span>,</span>
<span id="cb79-8"><a href="#cb79-8" aria-hidden="true" tabindex="-1"></a>                    <span class="st">'Tmem119'</span>,<span class="st">'Cx3cr1'</span>,</span>
<span id="cb79-9"><a href="#cb79-9" aria-hidden="true" tabindex="-1"></a>                    <span class="st">'Arg1'</span>,<span class="st">'Cd68'</span>)</span>
<span id="cb79-10"><a href="#cb79-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-11"><a href="#cb79-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-12"><a href="#cb79-12" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(sc.IC) <span class="ot">=</span> sc.IC<span class="sc">$</span>celltypesNew</span>
<span id="cb79-13"><a href="#cb79-13" aria-hidden="true" tabindex="-1"></a>celltypeDot <span class="ot">=</span> <span class="fu">DotPlot_scCustom</span>(sc.IC,<span class="at">features =</span> CellTypeMarkers,</span>
<span id="cb79-14"><a href="#cb79-14" aria-hidden="true" tabindex="-1"></a>                               <span class="at">x_lab_rotate =</span> <span class="cn">TRUE</span>,</span>
<span id="cb79-15"><a href="#cb79-15" aria-hidden="true" tabindex="-1"></a>                               <span class="at">dot.scale =</span> <span class="dv">5</span>,</span>
<span id="cb79-16"><a href="#cb79-16" aria-hidden="true" tabindex="-1"></a>                               <span class="at">flip_axes =</span> <span class="cn">TRUE</span>,</span>
<span id="cb79-17"><a href="#cb79-17" aria-hidden="true" tabindex="-1"></a>                               <span class="at">scale =</span> <span class="cn">TRUE</span>) <span class="sc">+</span> </span>
<span id="cb79-18"><a href="#cb79-18" aria-hidden="true" tabindex="-1"></a>                <span class="fu">scale_color_viridis</span>() <span class="sc">&amp;</span></span>
<span id="cb79-19"><a href="#cb79-19" aria-hidden="true" tabindex="-1"></a>                <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">0.12</span>,<span class="sc">-</span><span class="fl">0.25</span>),</span>
<span id="cb79-20"><a href="#cb79-20" aria-hidden="true" tabindex="-1"></a>                      <span class="at">legend.direction =</span> <span class="st">'horizontal'</span>,</span>
<span id="cb79-21"><a href="#cb79-21" aria-hidden="true" tabindex="-1"></a>                      <span class="at">legend.box =</span> <span class="st">'horizontal'</span>,</span>
<span id="cb79-22"><a href="#cb79-22" aria-hidden="true" tabindex="-1"></a>                      <span class="at">legend.key.size =</span> <span class="fu">unit</span>(<span class="fl">0.6</span>,<span class="st">'lines'</span>),</span>
<span id="cb79-23"><a href="#cb79-23" aria-hidden="true" tabindex="-1"></a>                      <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">6</span>),</span>
<span id="cb79-24"><a href="#cb79-24" aria-hidden="true" tabindex="-1"></a>                      <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>),</span>
<span id="cb79-25"><a href="#cb79-25" aria-hidden="true" tabindex="-1"></a>                      <span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">40</span>,<span class="dv">0</span>))</span>
<span id="cb79-26"><a href="#cb79-26" aria-hidden="true" tabindex="-1"></a>celltypeDot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/unnamed-chunk-47-1.png" class="img-fluid" width="480"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(figdir,<span class="st">'celltypes_dotplot.svg'</span>),celltypeDot,<span class="at">dpi =</span> <span class="dv">600</span>,<span class="at">width =</span> <span class="dv">11</span>,<span class="at">height =</span> <span class="fl">12.9</span>,<span class="at">units =</span> <span class="st">'cm'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a>celltypeFeature <span class="ot">=</span> <span class="fu">FeaturePlot</span>(sc.IC,</span>
<span id="cb81-2"><a href="#cb81-2" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">rev</span>(CellTypeMarkers),<span class="at">ncol =</span> <span class="dv">2</span>,</span>
<span id="cb81-3"><a href="#cb81-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">pt.size =</span> <span class="fl">0.1</span>) <span class="sc">&amp;</span> </span>
<span id="cb81-4"><a href="#cb81-4" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">NoAxes</span>() <span class="sc">&amp;</span> </span>
<span id="cb81-5"><a href="#cb81-5" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">scale_color_viridis</span>() <span class="sc">&amp;</span> </span>
<span id="cb81-6"><a href="#cb81-6" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'right'</span>, </span>
<span id="cb81-7"><a href="#cb81-7" aria-hidden="true" tabindex="-1"></a>                          <span class="at">legend.key.size =</span> <span class="fu">unit</span>(<span class="fl">0.3</span>,<span class="st">'lines'</span>),</span>
<span id="cb81-8"><a href="#cb81-8" aria-hidden="true" tabindex="-1"></a>                          <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">4</span>),</span>
<span id="cb81-9"><a href="#cb81-9" aria-hidden="true" tabindex="-1"></a>                          <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>))</span>
<span id="cb81-10"><a href="#cb81-10" aria-hidden="true" tabindex="-1"></a>celltypeFeature</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/cellTypeFeature-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(figdir,<span class="st">'celltypes_feature.svg'</span>),celltypeFeature,<span class="at">dpi =</span> <span class="dv">600</span>,<span class="at">width =</span> <span class="dv">7</span>,<span class="at">height =</span> <span class="fl">12.9</span>,<span class="at">units =</span> <span class="st">'cm'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="subcluster-analysis" class="level2">
<h2 class="anchored" data-anchor-id="subcluster-analysis">Subcluster analysis</h2>
<section id="macrophages-fig-x." class="level3">
<h3 class="anchored" data-anchor-id="macrophages-fig-x.">Macrophages Fig X.</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a>genes_mac <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb83-2"><a href="#cb83-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_general =</span> <span class="fu">c</span>(<span class="st">'Ly6c2'</span>, <span class="st">'Ccr2'</span>, <span class="st">'Top2a'</span>),</span>
<span id="cb83-3"><a href="#cb83-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_arg1 =</span> <span class="fu">c</span>(<span class="st">'Arg1'</span>,<span class="st">'Ecm1'</span>,<span class="st">'C1qb'</span>),</span>
<span id="cb83-4"><a href="#cb83-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_nos2 =</span> <span class="fu">c</span>(<span class="st">'Nos2'</span>,<span class="st">'Gpnmb'</span>,<span class="st">'Arg1'</span>),</span>
<span id="cb83-5"><a href="#cb83-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_cxcl10 =</span> <span class="fu">c</span>(<span class="st">'Cxcl10'</span>,<span class="st">'Cxcl9'</span>,<span class="st">'Il1b'</span>),</span>
<span id="cb83-6"><a href="#cb83-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_saa3 =</span> <span class="fu">c</span>(<span class="st">'Saa3'</span>,<span class="st">'Plac8'</span>,<span class="st">'Gbp2'</span>),</span>
<span id="cb83-7"><a href="#cb83-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_ifit1 =</span> <span class="fu">c</span>(<span class="st">'Ifit1'</span>,<span class="st">'Ifit2'</span>,<span class="st">'Ifit3'</span>)</span>
<span id="cb83-8"><a href="#cb83-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb83-9"><a href="#cb83-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-10"><a href="#cb83-10" aria-hidden="true" tabindex="-1"></a><span class="co">#give the order of the cluster levels </span></span>
<span id="cb83-11"><a href="#cb83-11" aria-hidden="true" tabindex="-1"></a>levels_mac <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Arg1+"</span>,<span class="st">"Nos2+"</span>,<span class="st">"Cxcl10+"</span>,<span class="st">"Saa3+"</span>,<span class="st">"Ifi2+"</span>)</span>
<span id="cb83-12"><a href="#cb83-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-13"><a href="#cb83-13" aria-hidden="true" tabindex="-1"></a><span class="co">#set the cluster levels to identiy of the macrophage object</span></span>
<span id="cb83-14"><a href="#cb83-14" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(mac.combined) <span class="ot">=</span> mac.combined<span class="sc">$</span>macTypes</span>
<span id="cb83-15"><a href="#cb83-15" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(mac.combined) <span class="ot">=</span> levels_mac</span>
<span id="cb83-16"><a href="#cb83-16" aria-hidden="true" tabindex="-1"></a>mac.combined<span class="sc">$</span>macTypes <span class="ot">=</span> <span class="fu">Idents</span>(mac.combined)</span>
<span id="cb83-17"><a href="#cb83-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-18"><a href="#cb83-18" aria-hidden="true" tabindex="-1"></a><span class="co">#plot the umaps split by sample</span></span>
<span id="cb83-19"><a href="#cb83-19" aria-hidden="true" tabindex="-1"></a>mac.cellStateUmap <span class="ot">=</span> <span class="fu">DimPlot_scCustom</span>(mac.combined,<span class="at">group.by =</span> <span class="st">'macTypes'</span>,<span class="at">label =</span> <span class="cn">TRUE</span>,<span class="at">split.by =</span> <span class="st">'sample'</span>)</span>
<span id="cb83-20"><a href="#cb83-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-21"><a href="#cb83-21" aria-hidden="true" tabindex="-1"></a><span class="co">#style the umap to have tiny arrows instead of axes </span></span>
<span id="cb83-22"><a href="#cb83-22" aria-hidden="true" tabindex="-1"></a>boundaries_mac <span class="ot">=</span> <span class="fu">list</span>(<span class="at">x =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">7</span>,<span class="dv">5</span>),<span class="at">y =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>,<span class="dv">5</span>))</span>
<span id="cb83-23"><a href="#cb83-23" aria-hidden="true" tabindex="-1"></a>arr <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">x =</span> boundaries_mac<span class="sc">$</span>x[<span class="dv">1</span>], <span class="at">y =</span> boundaries_mac<span class="sc">$</span>y[<span class="dv">1</span>], <span class="at">x_len =</span> <span class="dv">1</span>, <span class="at">y_len =</span> <span class="dv">2</span>,<span class="at">size =</span> <span class="dv">1</span>)</span>
<span id="cb83-24"><a href="#cb83-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-25"><a href="#cb83-25" aria-hidden="true" tabindex="-1"></a>mac.cellStateUmap[[<span class="dv">1</span>]] <span class="ot">=</span> mac.cellStateUmap[[<span class="dv">1</span>]] <span class="sc">+</span></span>
<span id="cb83-26"><a href="#cb83-26" aria-hidden="true" tabindex="-1"></a>                        theme_umap <span class="sc">+</span> </span>
<span id="cb83-27"><a href="#cb83-27" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">xlim</span>(boundaries_mac<span class="sc">$</span>x) <span class="sc">+</span> </span>
<span id="cb83-28"><a href="#cb83-28" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylim</span>(boundaries_mac<span class="sc">$</span>y) <span class="sc">+</span> </span>
<span id="cb83-29"><a href="#cb83-29" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">annotate</span>(<span class="st">"segment"</span>, </span>
<span id="cb83-30"><a href="#cb83-30" aria-hidden="true" tabindex="-1"></a>                             <span class="at">x =</span> arr<span class="sc">$</span>x, <span class="at">xend =</span> arr<span class="sc">$</span>x <span class="sc">+</span> <span class="fu">c</span>(arr<span class="sc">$</span>x_len, <span class="dv">0</span>), </span>
<span id="cb83-31"><a href="#cb83-31" aria-hidden="true" tabindex="-1"></a>                             <span class="at">y =</span> arr<span class="sc">$</span>y, <span class="at">yend =</span> arr<span class="sc">$</span>y <span class="sc">+</span> <span class="fu">c</span>(<span class="dv">0</span>, arr<span class="sc">$</span>y_len),</span>
<span id="cb83-32"><a href="#cb83-32" aria-hidden="true" tabindex="-1"></a>                             <span class="at">size =</span> arr<span class="sc">$</span>size,</span>
<span id="cb83-33"><a href="#cb83-33" aria-hidden="true" tabindex="-1"></a>                             <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">type =</span> <span class="st">"closed"</span>, <span class="at">length =</span> <span class="fu">unit</span>(<span class="dv">10</span>, <span class="st">'pt'</span>))) </span>
<span id="cb83-34"><a href="#cb83-34" aria-hidden="true" tabindex="-1"></a>mac.cellStateUmap[[<span class="dv">2</span>]] <span class="ot">=</span> mac.cellStateUmap[[<span class="dv">2</span>]] <span class="sc">+</span></span>
<span id="cb83-35"><a href="#cb83-35" aria-hidden="true" tabindex="-1"></a>                        theme_umap <span class="sc">+</span> </span>
<span id="cb83-36"><a href="#cb83-36" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">xlim</span>(boundaries_mac<span class="sc">$</span>x) <span class="sc">+</span> </span>
<span id="cb83-37"><a href="#cb83-37" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylim</span>(boundaries_mac<span class="sc">$</span>y) <span class="sc">+</span> </span>
<span id="cb83-38"><a href="#cb83-38" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">annotate</span>(<span class="st">"segment"</span>, </span>
<span id="cb83-39"><a href="#cb83-39" aria-hidden="true" tabindex="-1"></a>                             <span class="at">x =</span> arr<span class="sc">$</span>x, <span class="at">xend =</span> arr<span class="sc">$</span>x <span class="sc">+</span> <span class="fu">c</span>(arr<span class="sc">$</span>x_len, <span class="dv">0</span>), </span>
<span id="cb83-40"><a href="#cb83-40" aria-hidden="true" tabindex="-1"></a>                             <span class="at">y =</span> arr<span class="sc">$</span>y, <span class="at">yend =</span> arr<span class="sc">$</span>y <span class="sc">+</span> <span class="fu">c</span>(<span class="dv">0</span>, arr<span class="sc">$</span>y_len), </span>
<span id="cb83-41"><a href="#cb83-41" aria-hidden="true" tabindex="-1"></a>                             <span class="at">size =</span> arr<span class="sc">$</span>size,</span>
<span id="cb83-42"><a href="#cb83-42" aria-hidden="true" tabindex="-1"></a>                             <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">type =</span> <span class="st">"closed"</span>, <span class="at">length =</span> <span class="fu">unit</span>(<span class="dv">10</span>, <span class="st">'pt'</span>))) </span>
<span id="cb83-43"><a href="#cb83-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-44"><a href="#cb83-44" aria-hidden="true" tabindex="-1"></a><span class="co">#switch order of levels to have the plot read from top to bottom </span></span>
<span id="cb83-45"><a href="#cb83-45" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(mac.combined) <span class="ot">=</span> <span class="fu">rev</span>(levels_mac)</span>
<span id="cb83-46"><a href="#cb83-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-47"><a href="#cb83-47" aria-hidden="true" tabindex="-1"></a>mac.cellStateDot <span class="ot">=</span> <span class="fu">wrap_plots</span>(</span>
<span id="cb83-48"><a href="#cb83-48" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">lapply</span>(genes_mac,<span class="cf">function</span>(x){</span>
<span id="cb83-49"><a href="#cb83-49" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">DotPlot_scCustom</span>(mac.combined,</span>
<span id="cb83-50"><a href="#cb83-50" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">features =</span> x,</span>
<span id="cb83-51"><a href="#cb83-51" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">x_lab_rotate =</span> <span class="dv">90</span>,</span>
<span id="cb83-52"><a href="#cb83-52" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">y_lab_rotate =</span> <span class="dv">90</span>,</span>
<span id="cb83-53"><a href="#cb83-53" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">colors_use =</span> viridis_light_high) <span class="sc">+</span> </span>
<span id="cb83-54"><a href="#cb83-54" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'right'</span>,</span>
<span id="cb83-55"><a href="#cb83-55" aria-hidden="true" tabindex="-1"></a>                                <span class="at">legend.box =</span> <span class="st">'vertical'</span>,</span>
<span id="cb83-56"><a href="#cb83-56" aria-hidden="true" tabindex="-1"></a>                                <span class="at">legend.key.size =</span> <span class="fu">unit</span>(<span class="fl">0.75</span>,<span class="st">'line'</span>),</span>
<span id="cb83-57"><a href="#cb83-57" aria-hidden="true" tabindex="-1"></a>                                <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="fu">unit</span>(<span class="dv">8</span>,<span class="st">'pt'</span>)),</span>
<span id="cb83-58"><a href="#cb83-58" aria-hidden="true" tabindex="-1"></a>                                <span class="at">legend.title =</span> <span class="fu">element_blank</span>())     </span>
<span id="cb83-59"><a href="#cb83-59" aria-hidden="true" tabindex="-1"></a>                    }),<span class="at">ncol =</span> <span class="fu">length</span>(genes_mac))</span>
<span id="cb83-60"><a href="#cb83-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-61"><a href="#cb83-61" aria-hidden="true" tabindex="-1"></a>mac.ptable <span class="ot">=</span> <span class="fu">pTable</span>(mac.combined,<span class="st">'macTypes'</span>,<span class="st">'sample'</span>)</span>
<span id="cb83-62"><a href="#cb83-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-63"><a href="#cb83-63" aria-hidden="true" tabindex="-1"></a>mac.abundancePlot <span class="ot">=</span> mac.ptable<span class="sc">$</span>freq <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Freq,</span>
<span id="cb83-64"><a href="#cb83-64" aria-hidden="true" tabindex="-1"></a>                            <span class="at">y =</span> sample, </span>
<span id="cb83-65"><a href="#cb83-65" aria-hidden="true" tabindex="-1"></a>                            <span class="at">fill =</span> macTypes)) <span class="sc">+</span> </span>
<span id="cb83-66"><a href="#cb83-66" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">"identity"</span>) <span class="sc">+</span> </span>
<span id="cb83-67"><a href="#cb83-67" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb83-68"><a href="#cb83-68" aria-hidden="true" tabindex="-1"></a>                        <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb83-69"><a href="#cb83-69" aria-hidden="true" tabindex="-1"></a>                        <span class="at">axis.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb83-70"><a href="#cb83-70" aria-hidden="true" tabindex="-1"></a>                        <span class="at">panel.background =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span>  </span>
<span id="cb83-71"><a href="#cb83-71" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">paste0</span>(<span class="fu">trunc</span>(Freq),<span class="st">'%'</span>)), </span>
<span id="cb83-72"><a href="#cb83-72" aria-hidden="true" tabindex="-1"></a>                            <span class="at">position =</span> <span class="fu">position_stack</span>(<span class="at">vjust =</span> <span class="fl">0.5</span>),</span>
<span id="cb83-73"><a href="#cb83-73" aria-hidden="true" tabindex="-1"></a>                            <span class="at">size=</span><span class="dv">2</span>,</span>
<span id="cb83-74"><a href="#cb83-74" aria-hidden="true" tabindex="-1"></a>                            <span class="at">angle =</span> <span class="dv">90</span>) <span class="sc">+</span></span>
<span id="cb83-75"><a href="#cb83-75" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">palette36.colors</span>(<span class="at">n=</span><span class="dv">8</span>) <span class="sc">%&gt;%</span> <span class="fu">unname</span>()) </span>
<span id="cb83-76"><a href="#cb83-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-77"><a href="#cb83-77" aria-hidden="true" tabindex="-1"></a>fig_mac_sub <span class="ot">=</span> </span>
<span id="cb83-78"><a href="#cb83-78" aria-hidden="true" tabindex="-1"></a>mac.cellStateDot<span class="sc">/</span>mac.cellStateUmap<span class="sc">/</span>mac.abundancePlot<span class="sc">+</span><span class="fu">plot_layout</span>(<span class="at">design =</span> <span class="st">'A</span></span>
<span id="cb83-79"><a href="#cb83-79" aria-hidden="true" tabindex="-1"></a><span class="st">                                                                           B</span></span>
<span id="cb83-80"><a href="#cb83-80" aria-hidden="true" tabindex="-1"></a><span class="st">                                                                           B</span></span>
<span id="cb83-81"><a href="#cb83-81" aria-hidden="true" tabindex="-1"></a><span class="st">                                                                           C'</span>)</span>
<span id="cb83-82"><a href="#cb83-82" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(figdir,<span class="st">'fig_mac_sub.svg'</span>),<span class="at">plot =</span> fig_mac_sub,<span class="at">dpi =</span> <span class="dv">600</span>,<span class="at">width =</span> <span class="dv">50</span>,<span class="at">height =</span> <span class="dv">25</span>,<span class="at">units =</span> <span class="st">'cm'</span>)</span>
<span id="cb83-83"><a href="#cb83-83" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(figdir,<span class="st">'fig_mac_sub.svg'</span>),<span class="at">plot =</span> fig_mac_sub,<span class="at">dpi =</span> <span class="dv">600</span>,<span class="at">width =</span> <span class="dv">50</span>,<span class="at">height =</span> <span class="dv">25</span>,<span class="at">units =</span> <span class="st">'cm'</span>)</span>
<span id="cb83-84"><a href="#cb83-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-85"><a href="#cb83-85" aria-hidden="true" tabindex="-1"></a>fig_mac_sub</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/unnamed-chunk-48-1.png" class="img-fluid" width="1920"></p>
</div>
</div>
</section>
<section id="microglia-fig-x." class="level3">
<h3 class="anchored" data-anchor-id="microglia-fig-x.">Microglia Fig X.</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a>genes_MG <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_hMG =</span> <span class="fu">c</span>(<span class="st">'Gpr34'</span>,<span class="st">'Siglech'</span>,<span class="st">'P2ry12'</span>,<span class="st">'Tmem119'</span>),</span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_daMG2 =</span> <span class="fu">c</span>(<span class="st">'Ctsb'</span>,<span class="st">'Apoe'</span>,<span class="st">'B2m'</span>,<span class="st">'Cst7'</span>,<span class="st">'Cd74'</span>,<span class="st">'Ly86'</span>),</span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_daMG3 =</span> <span class="fu">c</span>(<span class="st">'Cxcl10'</span>,<span class="st">'Tnf'</span>,<span class="st">'Ccl4'</span>,<span class="st">'Il1a'</span>,<span class="st">'B2m'</span>,<span class="st">'Ly86'</span>),</span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes_repMG =</span> <span class="fu">c</span>(<span class="st">'Top2a'</span>,<span class="st">'Mki67'</span>))</span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-7"><a href="#cb84-7" aria-hidden="true" tabindex="-1"></a><span class="co">#give the order of the cluster levels </span></span>
<span id="cb84-8"><a href="#cb84-8" aria-hidden="true" tabindex="-1"></a>levels_mg <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'hMG'</span>,<span class="st">'hMG/daMG'</span>,<span class="st">'daMG2'</span>,<span class="st">'daMG3'</span>,<span class="st">'replicating MG'</span>)</span>
<span id="cb84-9"><a href="#cb84-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-10"><a href="#cb84-10" aria-hidden="true" tabindex="-1"></a><span class="co">#set the cluster levels to identiy of the macrophage object</span></span>
<span id="cb84-11"><a href="#cb84-11" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(mg.combined) <span class="ot">=</span> mg.combined<span class="sc">$</span>mgTypes</span>
<span id="cb84-12"><a href="#cb84-12" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(mg.combined) <span class="ot">=</span> levels_mg</span>
<span id="cb84-13"><a href="#cb84-13" aria-hidden="true" tabindex="-1"></a>mg.combined<span class="sc">$</span>mgTypes <span class="ot">=</span> <span class="fu">Idents</span>(mg.combined)</span>
<span id="cb84-14"><a href="#cb84-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-15"><a href="#cb84-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-16"><a href="#cb84-16" aria-hidden="true" tabindex="-1"></a><span class="co">#generate umap of mg subtypes</span></span>
<span id="cb84-17"><a href="#cb84-17" aria-hidden="true" tabindex="-1"></a>mg.cellStateUmap <span class="ot">=</span> <span class="fu">DimPlot_scCustom</span>(mg.combined,<span class="at">label =</span> <span class="cn">TRUE</span>,<span class="at">split.by =</span> <span class="st">'sample'</span>) </span>
<span id="cb84-18"><a href="#cb84-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-19"><a href="#cb84-19" aria-hidden="true" tabindex="-1"></a><span class="co">#style the umaps </span></span>
<span id="cb84-20"><a href="#cb84-20" aria-hidden="true" tabindex="-1"></a>boundaries_mg <span class="ot">=</span> <span class="fu">list</span>(<span class="at">x =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">7.5</span>,<span class="dv">5</span>),<span class="at">y =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">4.5</span>,<span class="dv">6</span>))</span>
<span id="cb84-21"><a href="#cb84-21" aria-hidden="true" tabindex="-1"></a>arr <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">x =</span> boundaries_mg<span class="sc">$</span>x[<span class="dv">1</span>], <span class="at">y =</span> boundaries_mg<span class="sc">$</span>y[<span class="dv">1</span>], <span class="at">x_len =</span> <span class="dv">1</span>, <span class="at">y_len =</span> <span class="dv">2</span>,<span class="at">size =</span> <span class="dv">1</span>)</span>
<span id="cb84-22"><a href="#cb84-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-23"><a href="#cb84-23" aria-hidden="true" tabindex="-1"></a>mg.cellStateUmap[[<span class="dv">1</span>]] <span class="ot">=</span> mg.cellStateUmap[[<span class="dv">1</span>]] <span class="sc">+</span></span>
<span id="cb84-24"><a href="#cb84-24" aria-hidden="true" tabindex="-1"></a>                        theme_umap <span class="sc">+</span> </span>
<span id="cb84-25"><a href="#cb84-25" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">xlim</span>(boundaries_mg<span class="sc">$</span>x) <span class="sc">+</span> </span>
<span id="cb84-26"><a href="#cb84-26" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylim</span>(boundaries_mg<span class="sc">$</span>y) <span class="sc">+</span> </span>
<span id="cb84-27"><a href="#cb84-27" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">annotate</span>(<span class="st">"segment"</span>, </span>
<span id="cb84-28"><a href="#cb84-28" aria-hidden="true" tabindex="-1"></a>                             <span class="at">x =</span> arr<span class="sc">$</span>x, <span class="at">xend =</span> arr<span class="sc">$</span>x <span class="sc">+</span> <span class="fu">c</span>(arr<span class="sc">$</span>x_len, <span class="dv">0</span>), </span>
<span id="cb84-29"><a href="#cb84-29" aria-hidden="true" tabindex="-1"></a>                             <span class="at">y =</span> arr<span class="sc">$</span>y, <span class="at">yend =</span> arr<span class="sc">$</span>y <span class="sc">+</span> <span class="fu">c</span>(<span class="dv">0</span>, arr<span class="sc">$</span>y_len), </span>
<span id="cb84-30"><a href="#cb84-30" aria-hidden="true" tabindex="-1"></a>                             <span class="at">size =</span> arr<span class="sc">$</span>size,</span>
<span id="cb84-31"><a href="#cb84-31" aria-hidden="true" tabindex="-1"></a>                             <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">type =</span> <span class="st">"closed"</span>, <span class="at">length =</span> <span class="fu">unit</span>(<span class="dv">10</span>, <span class="st">'pt'</span>))) </span>
<span id="cb84-32"><a href="#cb84-32" aria-hidden="true" tabindex="-1"></a>mg.cellStateUmap[[<span class="dv">2</span>]] <span class="ot">=</span> mg.cellStateUmap[[<span class="dv">2</span>]] <span class="sc">+</span></span>
<span id="cb84-33"><a href="#cb84-33" aria-hidden="true" tabindex="-1"></a>                        theme_umap <span class="sc">+</span> </span>
<span id="cb84-34"><a href="#cb84-34" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">xlim</span>(boundaries_mg<span class="sc">$</span>x) <span class="sc">+</span> </span>
<span id="cb84-35"><a href="#cb84-35" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylim</span>(boundaries_mg<span class="sc">$</span>y) <span class="sc">+</span> </span>
<span id="cb84-36"><a href="#cb84-36" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">annotate</span>(<span class="st">"segment"</span>, </span>
<span id="cb84-37"><a href="#cb84-37" aria-hidden="true" tabindex="-1"></a>                             <span class="at">x =</span> arr<span class="sc">$</span>x, <span class="at">xend =</span> arr<span class="sc">$</span>x <span class="sc">+</span> <span class="fu">c</span>(arr<span class="sc">$</span>x_len, <span class="dv">0</span>), </span>
<span id="cb84-38"><a href="#cb84-38" aria-hidden="true" tabindex="-1"></a>                             <span class="at">y =</span> arr<span class="sc">$</span>y, <span class="at">yend =</span> arr<span class="sc">$</span>y <span class="sc">+</span> <span class="fu">c</span>(<span class="dv">0</span>, arr<span class="sc">$</span>y_len), </span>
<span id="cb84-39"><a href="#cb84-39" aria-hidden="true" tabindex="-1"></a>                             <span class="at">size =</span> arr<span class="sc">$</span>size,</span>
<span id="cb84-40"><a href="#cb84-40" aria-hidden="true" tabindex="-1"></a>                             <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">type =</span> <span class="st">"closed"</span>, <span class="at">length =</span> <span class="fu">unit</span>(<span class="dv">10</span>, <span class="st">'pt'</span>))) </span>
<span id="cb84-41"><a href="#cb84-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-42"><a href="#cb84-42" aria-hidden="true" tabindex="-1"></a><span class="co">#reverse cluster levels for dotplot to read from top to bottom </span></span>
<span id="cb84-43"><a href="#cb84-43" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(mg.combined) <span class="ot">=</span> <span class="fu">rev</span>(levels_mg)</span>
<span id="cb84-44"><a href="#cb84-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-45"><a href="#cb84-45" aria-hidden="true" tabindex="-1"></a>mg.cellStateDot <span class="ot">=</span>  <span class="fu">wrap_plots</span>(</span>
<span id="cb84-46"><a href="#cb84-46" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">lapply</span>(genes_MG,<span class="cf">function</span>(x){</span>
<span id="cb84-47"><a href="#cb84-47" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">DotPlot_scCustom</span>(mg.combined,</span>
<span id="cb84-48"><a href="#cb84-48" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">features =</span> x,</span>
<span id="cb84-49"><a href="#cb84-49" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">x_lab_rotate =</span> <span class="dv">90</span>,</span>
<span id="cb84-50"><a href="#cb84-50" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">y_lab_rotate =</span> <span class="dv">90</span>,</span>
<span id="cb84-51"><a href="#cb84-51" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">colors_use =</span> viridis_light_high) <span class="sc">+</span> </span>
<span id="cb84-52"><a href="#cb84-52" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'right'</span>,</span>
<span id="cb84-53"><a href="#cb84-53" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">legend.box =</span> <span class="st">'vertical'</span>,</span>
<span id="cb84-54"><a href="#cb84-54" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">legend.key.size =</span> <span class="fu">unit</span>(<span class="fl">0.75</span>,<span class="st">'line'</span>),</span>
<span id="cb84-55"><a href="#cb84-55" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="fu">unit</span>(<span class="dv">8</span>,<span class="st">'pt'</span>)),</span>
<span id="cb84-56"><a href="#cb84-56" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">legend.title =</span> <span class="fu">element_blank</span>())</span>
<span id="cb84-57"><a href="#cb84-57" aria-hidden="true" tabindex="-1"></a>                      }), <span class="at">ncol =</span> <span class="fu">length</span>(genes_MG))</span>
<span id="cb84-58"><a href="#cb84-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-59"><a href="#cb84-59" aria-hidden="true" tabindex="-1"></a>mg.cellStateDot <span class="ot">=</span> mg.cellStateDot <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'right'</span>,</span>
<span id="cb84-60"><a href="#cb84-60" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">legend.box =</span> <span class="st">'vertical'</span>,</span>
<span id="cb84-61"><a href="#cb84-61" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">legend.key.size =</span> <span class="fu">unit</span>(<span class="fl">0.75</span>,<span class="st">'line'</span>),</span>
<span id="cb84-62"><a href="#cb84-62" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="fu">unit</span>(<span class="dv">8</span>,<span class="st">'pt'</span>)),</span>
<span id="cb84-63"><a href="#cb84-63" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">legend.title =</span> <span class="fu">element_blank</span>())</span>
<span id="cb84-64"><a href="#cb84-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-65"><a href="#cb84-65" aria-hidden="true" tabindex="-1"></a>mg.ptable <span class="ot">=</span> <span class="fu">pTable</span>(mg.combined,<span class="st">'mgTypes'</span>,<span class="st">'sample'</span>)</span>
<span id="cb84-66"><a href="#cb84-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-67"><a href="#cb84-67" aria-hidden="true" tabindex="-1"></a>mg.abundancePlot <span class="ot">=</span> mg.ptable<span class="sc">$</span>freq <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Freq,</span>
<span id="cb84-68"><a href="#cb84-68" aria-hidden="true" tabindex="-1"></a>                            <span class="at">y =</span> sample, </span>
<span id="cb84-69"><a href="#cb84-69" aria-hidden="true" tabindex="-1"></a>                            <span class="at">fill =</span> mgTypes)) <span class="sc">+</span> </span>
<span id="cb84-70"><a href="#cb84-70" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">"identity"</span>) <span class="sc">+</span> </span>
<span id="cb84-71"><a href="#cb84-71" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb84-72"><a href="#cb84-72" aria-hidden="true" tabindex="-1"></a>                        <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb84-73"><a href="#cb84-73" aria-hidden="true" tabindex="-1"></a>                        <span class="at">axis.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb84-74"><a href="#cb84-74" aria-hidden="true" tabindex="-1"></a>                        <span class="at">panel.background =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span>  </span>
<span id="cb84-75"><a href="#cb84-75" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">paste0</span>(<span class="fu">trunc</span>(Freq),<span class="st">'%'</span>)), </span>
<span id="cb84-76"><a href="#cb84-76" aria-hidden="true" tabindex="-1"></a>                            <span class="at">position =</span> <span class="fu">position_stack</span>(<span class="at">vjust =</span> <span class="fl">0.5</span>),</span>
<span id="cb84-77"><a href="#cb84-77" aria-hidden="true" tabindex="-1"></a>                            <span class="at">size=</span><span class="dv">2</span>,</span>
<span id="cb84-78"><a href="#cb84-78" aria-hidden="true" tabindex="-1"></a>                            <span class="at">angle =</span> <span class="dv">90</span>) <span class="sc">+</span></span>
<span id="cb84-79"><a href="#cb84-79" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">palette36.colors</span>(<span class="at">n=</span><span class="dv">8</span>) <span class="sc">%&gt;%</span> <span class="fu">unname</span>()) </span>
<span id="cb84-80"><a href="#cb84-80" aria-hidden="true" tabindex="-1"></a>fig_mg_sub <span class="ot">=</span> </span>
<span id="cb84-81"><a href="#cb84-81" aria-hidden="true" tabindex="-1"></a>  mg.cellStateDot<span class="sc">/</span></span>
<span id="cb84-82"><a href="#cb84-82" aria-hidden="true" tabindex="-1"></a>  mg.cellStateUmap<span class="sc">/</span></span>
<span id="cb84-83"><a href="#cb84-83" aria-hidden="true" tabindex="-1"></a>  mg.abundancePlot <span class="sc">+</span> </span>
<span id="cb84-84"><a href="#cb84-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_layout</span>(<span class="at">design =</span> <span class="st">'A</span></span>
<span id="cb84-85"><a href="#cb84-85" aria-hidden="true" tabindex="-1"></a><span class="st">                        B</span></span>
<span id="cb84-86"><a href="#cb84-86" aria-hidden="true" tabindex="-1"></a><span class="st">                        B</span></span>
<span id="cb84-87"><a href="#cb84-87" aria-hidden="true" tabindex="-1"></a><span class="st">                        C'</span>)</span>
<span id="cb84-88"><a href="#cb84-88" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(figdir,<span class="st">'fig_mg_sub.svg'</span>),<span class="at">plot =</span> fig_mg_sub,<span class="at">dpi =</span> <span class="dv">600</span>,<span class="at">width =</span> <span class="dv">50</span>,<span class="at">height =</span> <span class="dv">25</span>,<span class="at">units =</span> <span class="st">'cm'</span>)</span>
<span id="cb84-89"><a href="#cb84-89" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(figdir,<span class="st">'fig_mg_sub.svg'</span>),<span class="at">plot =</span> fig_mg_sub,<span class="at">dpi =</span> <span class="dv">600</span>,<span class="at">width =</span> <span class="dv">50</span>,<span class="at">height =</span> <span class="dv">25</span>,<span class="at">units =</span> <span class="st">'cm'</span>)</span>
<span id="cb84-90"><a href="#cb84-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-91"><a href="#cb84-91" aria-hidden="true" tabindex="-1"></a>fig_mg_sub</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="EAE_IC_files/figure-html/unnamed-chunk-49-1.png" class="img-fluid" width="1920"></p>
</div>
</div>
</section>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>